diff -Naur indicator-printers-0.1.6/src/indicator-printers-service.c indicator-printers-0.1.6.patched/src/indicator-printers-service.c
--- indicator-printers-0.1.6/src/indicator-printers-service.c	2012-04-04 21:47:41.000000000 +0200
+++ indicator-printers-0.1.6.patched/src/indicator-printers-service.c	2013-04-04 16:05:15.185305620 +0200
@@ -56,7 +56,7 @@
 
     attr = ippFindAttribute (resp, "notify-subscription-id", IPP_TAG_INTEGER);
     if (attr)
-        id = attr->values[0].integer;
+        id = ippGetInteger (attr, 0);
     else
         g_warning ("ipp-create-printer-subscription response doesn't contain "
                    "subscription id.\n");
