=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2012-01-29 18:17:56 +0000
+++ CMakeLists.txt	2012-03-08 08:40:22 +0000
@@ -15,37 +15,13 @@
 add_definitions(-DGETTEXT_PACKAGE=\"${GETTEXT_PACKAGE}\")
 
 # GSettings Schema
-
-SET (MAYA_SCHEMAS "org.gnome.euclide.gschema.xml")
-# Have an option to not install the schema into where GLib is
-
-option (GSETTINGS_LOCALINSTALL "Install GSettings Schemas locally instead of to the GLib prefix" OFF)
-if (GSETTINGS_LOCALINSTALL)
-	
-
-    SET (GSETTINGS_DIR "${CMAKE_INSTALL_PREFIX}/share/glib-2.0/schemas/")
-else (GSETTINGS_LOCALINSTALL)
-execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} glib-2.0 --variable prefix OUTPUT_VARIABLE _glib_prefix OUTPUT_STRIP_TRAILING_WHITESPACE)
-SET (GSETTINGS_DIR "${_glib_prefix}/share/glib-2.0/schemas/")
-endif (GSETTINGS_LOCALINSTALL)
-# Run the validator and error if it fails
-
-execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} gio-2.0 --variable glib_compile_schemas  OUTPUT_VARIABLE _glib_comple_schemas OUTPUT_STRIP_TRAILING_WHITESPACE)
-execute_process (COMMAND ${_glib_comple_schemas} --dry-run --schema-file=${CMAKE_CURRENT_SOURCE_DIR}/${MAYA_SCHEMAS} ERROR_VARIABLE _schemas_invalid OUTPUT_STRIP_TRAILING_WHITESPACE)
-if (_schemas_invalid)
-message (SEND_ERROR "Schema validation error: ${_schemas_invalid}")
-endif (_schemas_invalid)
-# Actually install and recomple schemas
-
-message (STATUS "GSettings schemas will be installed into ${GSETTINGS_DIR}")
-install (FILES ${MAYA_SCHEMAS} DESTINATION ${GSETTINGS_DIR} OPTIONAL)
-install (CODE "message (STATUS \"Compiling GSettings schemas\")
-")
-install (CODE "execute_process (COMMAND ${_glib_comple_schemas} ${GSETTINGS_DIR})
-")
+include (GSettings)
+add_schema ("org.gnome.euclide.gschema.xml")
+
+# Process subdirectories
 add_subdirectory(euclidecore)
 add_subdirectory(data)
 add_subdirectory(completion)
 add_subdirectory(symbols)
+add_subdirectory(po)
 
-add_subdirectory(po)
\ No newline at end of file

=== added file 'cmake/Tests.cmake'
--- cmake/Tests.cmake	1970-01-01 00:00:00 +0000
+++ cmake/Tests.cmake	2012-03-08 08:40:22 +0000
@@ -0,0 +1,5 @@
+# Test macros for Marlin, feel free to re-use them.
+
+macro(add_test_executable EXE_NAME)
+    add_test(${EXE_NAME} gtester ${CMAKE_CURRENT_BINARY_DIR}/${EXE_NAME})
+endmacro()

=== modified file 'cmake/Translations.cmake'
--- cmake/Translations.cmake	2011-10-31 17:16:51 +0000
+++ cmake/Translations.cmake	2012-03-08 08:40:22 +0000
@@ -36,6 +36,6 @@
 
     add_custom_command (TARGET pot COMMAND
         ${XGETTEXT_EXECUTABLE} -d ${NLS_PACKAGE} -o ${CMAKE_CURRENT_SOURCE_DIR}/${NLS_PACKAGE}.pot
-        ${VALA_SOURCE} ${C_SOURCE} --keyword="_" --from-code=UTF-8
+        ${VALA_SOURCE} ${C_SOURCE} --keyword="_" --keyword="N_" --from-code=UTF-8
         )
 endmacro()

=== modified file 'cmake/ValaPrecompile.cmake'
--- cmake/ValaPrecompile.cmake	2011-10-31 17:16:51 +0000
+++ cmake/ValaPrecompile.cmake	2012-03-08 08:40:22 +0000
@@ -117,6 +117,7 @@
     set(out_files "")
     set(${output} "")
     foreach(src ${ARGS_DEFAULT_ARGS})
+        string(REPLACE ${CMAKE_CURRENT_SOURCE_DIR}/ "" src ${src})
         string(REGEX MATCH "^/" IS_MATCHED ${src})
         if(${IS_MATCHED} MATCHES "/")
             list(APPEND in_files "${src}")

