=== modified file 'src/main.vala'
--- src/main.vala	2012-05-18 15:08:15 +0000
+++ src/main.vala	2012-05-18 15:39:16 +0000
@@ -542,7 +542,7 @@
 
     void on_show_popover()
     {
-        build_arrow.set_state(true);
+        build_arrow.set_state(Gtk.StateType.NORMAL);
         var build_popover = new Granite.Widgets.PopOver();
         build_popover.move_to_widget(build_arrow);
         var box = build_popover.get_content_area() as Gtk.Box;
@@ -559,7 +559,7 @@
         build_popover.run();
         cmake.arguments = arguments.text;
         build_popover.destroy();
-        build_arrow.set_state(false);
+        build_arrow.set_state(Gtk.StateType.INSENSITIVE);
         
     }
     

