=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2013-12-26 00:08:04 +0000
+++ CMakeLists.txt	2014-01-31 17:57:40 +0000
@@ -34,8 +34,28 @@
 
 # Slingshot
 
-set (CORE_DEPS "gobject-2.0;glib-2.0;gio-2.0;gio-unix-2.0;gee-1.0;libgnome-menu;libwnck-3.0;gdk-x11-3.0;unity;")
-set (UI_DEPS "gtk+-3.0>=3.2.0;granite;zeitgeist-2.0;")
+option (USE_UNITY "Use Unity integration" ON)
+
+if (USE_UNITY)
+  message ("-- Unity integration enabled")
+  set (UNITY_DEPS unity)
+  set (UNITY_OPTIONS --define=HAVE_UNITY)
+else ()
+  message ("-- Unity integration disabled")
+endif ()
+
+option (USE_ZEITGEIST "Use Zeitgeist integration" ON)
+
+if (USE_ZEITGEIST)
+  message ("-- Zeitgeist integration enabled")
+  set (ZEITGEIST_DEPS zeitgeist-2.0)
+  set (ZEITGEIST_OPTIONS --define=HAVE_ZEITGEIST)
+else ()
+  message ("-- Zeitgeist integration disabled")
+endif ()
+
+set (CORE_DEPS "gobject-2.0;glib-2.0;gio-2.0;gio-unix-2.0;gee-1.0;libgnome-menu;libwnck-3.0;gdk-x11-3.0;${UNITY_DEPS};")
+set (UI_DEPS "gtk+-3.0>=3.2.0;granite;${ZEITGEIST_DEPS};")
 
 find_package (PkgConfig)
 pkg_check_modules (DEPS REQUIRED "${CORE_DEPS}${UI_DEPS}" gthread-2.0)
@@ -69,6 +89,8 @@
 OPTIONS
     --thread
     -g
+    ${UNITY_OPTIONS}
+    ${ZEITGEIST_OPTIONS}
 )
 
 # Comment this out to enable C compiler warnings
=== modified file 'src/Backend/App.vala'
--- src/Backend/App.vala	2013-12-26 00:08:04 +0000
+++ src/Backend/App.vala	2014-01-31 17:57:40 +0000
@@ -42,7 +42,9 @@
         desktop_id = entry.get_desktop_file_id ();
         icon_name = entry.get_icon () ?? "application-default-icon";
         desktop_path = entry.get_desktop_file_path ();
+#if HAVE_UNITY
         keywords = Unity.AppInfoManager.get_default ().get_keywords (desktop_id);
+#endif
         generic_name = entry.get_generic_name ();
 
         update_icon ();

=== modified file 'src/Backend/AppSystem.vala'
--- src/Backend/AppSystem.vala	2013-12-26 00:08:04 +0000
+++ src/Backend/AppSystem.vala	2014-01-31 15:20:32 +0000
@@ -22,15 +22,19 @@
     private Gee.HashMap<string, Gee.ArrayList<App>> apps = null;
     private GMenu.Tree apps_menu = null;
 
+#if HAVE_ZEITGEIST
     private RelevancyService rl_service;
+#endif
 
     public signal void changed ();
     private bool index_changed = false;
 
     construct {
 
+#if HAVE_ZEITGEIST
         rl_service = new RelevancyService ();
         rl_service.update_complete.connect (update_popularity);
+#endif
 
         apps_menu = GMenu.Tree.lookup ("pantheon-applications.menu", GMenu.TreeFlags.INCLUDE_EXCLUDED);
         apps_menu.add_monitor ((menu) => {
@@ -49,7 +53,9 @@
 
     private void update_app_system () {
 
+#if HAVE_ZEITGEIST
         rl_service.refresh_popularity ();
+#endif
 
         update_categories_index ();
         update_apps ();
@@ -72,12 +78,14 @@
 
     }
 
+#if HAVE_ZEITGEIST
     private void update_popularity () {
 
         foreach (Gee.ArrayList<App> category in apps.values)
             foreach (App app in category)
                 app.popularity = rl_service.get_app_popularity (app.desktop_id);
     }
+#endif
 
     private void update_apps () {
 
@@ -118,7 +126,9 @@
                 case GMenu.TreeItemType.ENTRY:
                     if (is_entry ((GMenu.TreeEntry) item)) {
                         app = new App ((GMenu.TreeEntry) item);
+#if HAVE_ZEITGEIST
                         app.launched.connect (rl_service.app_launched);
+#endif
                         app_list.add (app);
                     }
                     break;

=== modified file 'src/Backend/RelevancyService.vala'
--- src/Backend/RelevancyService.vala	2013-12-26 00:08:04 +0000
+++ src/Backend/RelevancyService.vala	2014-01-31 15:20:32 +0000
@@ -17,6 +17,7 @@
 //
 //  Thanks to Synapse Developers for this class
 
+#if HAVE_ZEITGEIST
 public class Slingshot.Backend.RelevancyService : Object {
 
     private Zeitgeist.Log zg_log;
@@ -177,4 +178,5 @@
             critical (e.message);
         }
     }
-}
\ No newline at end of file
+}
+#endif

