=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2014-01-31 15:20:32 +0000
+++ CMakeLists.txt	2014-01-31 17:52:40 +0000
@@ -34,6 +34,16 @@
 
 # Slingshot
 
+option (USE_UNITY "Use Unity integration" ON)
+
+if (USE_UNITY)
+  message ("-- Unity integration enabled")
+  set (UNITY_DEPS unity)
+  set (UNITY_OPTIONS --define=HAVE_UNITY)
+else ()
+  message ("-- Unity integration disabled")
+endif ()
+
 option (USE_ZEITGEIST "Use Zeitgeist integration" ON)
 
 if (USE_ZEITGEIST)
@@ -44,7 +54,7 @@
   message ("-- Zeitgeist integration disabled")
 endif ()
 
-set (CORE_DEPS "gobject-2.0;glib-2.0;gio-2.0;gio-unix-2.0;gee-1.0;libgnome-menu;libwnck-3.0;gdk-x11-3.0;unity;")
+set (CORE_DEPS "gobject-2.0;glib-2.0;gio-2.0;gio-unix-2.0;gee-1.0;libgnome-menu;libwnck-3.0;gdk-x11-3.0;${UNITY_DEPS};")
 set (UI_DEPS "gtk+-3.0>=3.2.0;granite;${ZEITGEIST_DEPS};")
 
 find_package (PkgConfig)
@@ -79,6 +89,7 @@
 OPTIONS
     --thread
     -g
+    ${UNITY_OPTIONS}
     ${ZEITGEIST_OPTIONS}
 )
 

=== modified file 'src/Backend/App.vala'
--- src/Backend/App.vala	2013-12-26 00:08:04 +0000
+++ src/Backend/App.vala	2014-01-31 17:49:58 +0000
@@ -42,7 +42,9 @@
         desktop_id = entry.get_desktop_file_id ();
         icon_name = entry.get_icon () ?? "application-default-icon";
         desktop_path = entry.get_desktop_file_path ();
+#if HAVE_UNITY
         keywords = Unity.AppInfoManager.get_default ().get_keywords (desktop_id);
+#endif
         generic_name = entry.get_generic_name ();
 
         update_icon ();

