=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2013-12-01 14:35:04 +0000
+++ CMakeLists.txt	2014-01-24 18:17:47 +0000
@@ -56,7 +56,7 @@
 set (CORE_PACKAGES
   glib-2.0
   gio-2.0
-  gee-1.0
+  gee-0.8
   libpeas-1.0
   libpeas-gtk-1.0
   gtk+-3.0
@@ -69,7 +69,7 @@
 set (CORE_PKG
   glib-2.0>=2.32
   gio-2.0
-  gee-1.0
+  gee-0.8
   libpeas-1.0
   libpeas-gtk-1.0
   gtk+-3.0>=3.4.2

=== modified file 'core/noise-core.deps'
--- core/noise-core.deps	2013-06-02 20:07:16 +0000
+++ core/noise-core.deps	2014-01-24 18:17:47 +0000
@@ -1,6 +1,6 @@
 glib-2.0
 gio-2.0
-gee-1.0
+gee-0.8
 libpeas-1.0
 libpeas-gtk-1.0
 gtk+-3.0

=== modified file 'core/noise-core.pc.cmake'
--- core/noise-core.pc.cmake	2013-06-12 18:04:28 +0000
+++ core/noise-core.pc.cmake	2014-01-24 18:17:47 +0000
@@ -8,5 +8,5 @@
 Version: 0.2  
 Libs: -lnoise-core
 Cflags: -I@DOLLAR@{includedir}/noise-core
-Requires: glib-2.0 gio-2.0 gee-1.0 libpeas-1.0 libpeas-gtk-1.0 gtk+-3.0 granite gstreamer-0.10 gstreamer-interfaces-0.10 gstreamer-tag-0.10 gstreamer-pbutils-0.10
+Requires: glib-2.0 gio-2.0 gee-0.8 libpeas-1.0 libpeas-gtk-1.0 gtk+-3.0 granite gstreamer-0.10 gstreamer-interfaces-0.10 gstreamer-tag-0.10 gstreamer-pbutils-0.10
 

=== modified file 'src/Dialogs/TransferFromDeviceDialog.vala'
--- src/Dialogs/TransferFromDeviceDialog.vala	2013-02-28 23:45:33 +0000
+++ src/Dialogs/TransferFromDeviceDialog.vala	2014-01-24 18:17:47 +0000
@@ -145,7 +145,7 @@
 		var medias_sorted = new LinkedList<Media>();
 		foreach(var m in medias)
 			medias_sorted.add(m);
-		medias_sorted.sort((CompareFunc)mediaCompareFunc);
+		medias_sorted.sort(mediaCompareFunc);
 		
 		foreach(var s in medias_sorted) {
 			TreeIter item;

