diff -Naur indicator-power-12.10.6+14.04.20131129.orig/configure.ac indicator-power-12.10.6+14.04.20131129/configure.ac
--- indicator-power-12.10.6+14.04.20131129.orig/configure.ac	2013-12-03 15:02:38.831155608 +0100
+++ indicator-power-12.10.6+14.04.20131129/configure.ac	2013-12-03 15:03:17.924395447 +0100
@@ -40,8 +40,7 @@
 PKG_CHECK_MODULES([SERVICE_DEPS],[glib-2.0 >= $GLIB_REQUIRED_VERSION
                                   gio-2.0 >= $GIO_REQUIRED_VERSION
                                   gio-unix-2.0 >= $GIO_UNIX_REQUIRED_VERSION
-                                  gudev-1.0 >= $GUDEV_REQUIRED_VERSION
-                                  url-dispatcher-1])
+                                  gudev-1.0 >= $GUDEV_REQUIRED_VERSION])
 
 ###########################
 # GSETTINGS
diff -Naur indicator-power-12.10.6+14.04.20131129.orig/src/service.c indicator-power-12.10.6+14.04.20131129/src/service.c
--- indicator-power-12.10.6+14.04.20131129.orig/src/service.c	2013-12-03 15:02:38.832155615 +0100
+++ indicator-power-12.10.6+14.04.20131129/src/service.c	2013-12-03 15:04:00.205654570 +0100
@@ -22,7 +22,6 @@
 
 #include <glib/gi18n.h>
 #include <gio/gio.h>
-#include <url-dispatcher.h>
 
 #include "device.h"
 #include "device-provider.h"
@@ -701,14 +700,6 @@
   execute_command ("gnome-power-statistics");
 }
 
-static void
-on_phone_settings_activated (GSimpleAction * a      G_GNUC_UNUSED,
-                             GVariant      * param  G_GNUC_UNUSED,
-                             gpointer        gself  G_GNUC_UNUSED)
-{
-  url_dispatch_send("settings:///system/battery", NULL, NULL);
-}
-
 /***
 ****
 ***/
@@ -723,7 +714,6 @@
 
   GActionEntry entries[] = {
     { "activate-settings", on_settings_activated },
-    { "activate-phone-settings", on_phone_settings_activated },
     { "activate-statistics", on_statistics_activated }
   };
 
