diff -Naur indicator-power-12.10.6+14.10.20141015-orig/data/CMakeLists.txt indicator-power-12.10.6+14.10.20141015/data/CMakeLists.txt
--- indicator-power-12.10.6+14.10.20141015-orig/data/CMakeLists.txt	2015-04-13 21:51:10.041032911 +0200
+++ indicator-power-12.10.6+14.10.20141015/data/CMakeLists.txt	2015-04-13 21:52:09.880430436 +0200
@@ -15,26 +15,6 @@
 add_schema (${SCHEMA_FILE})
 
 ##
-##  Upstart Job File
-##
-
-# where to install
-set (UPSTART_JOB_DIR "${CMAKE_INSTALL_FULL_DATADIR}/upstart/sessions")
-message (STATUS "${UPSTART_JOB_DIR} is the Upstart Job File install dir")
-
-set (UPSTART_JOB_NAME "${CMAKE_PROJECT_NAME}.conf")
-set (UPSTART_JOB_FILE "${CMAKE_CURRENT_BINARY_DIR}/${UPSTART_JOB_NAME}")
-set (UPSTART_JOB_FILE_IN "${CMAKE_CURRENT_SOURCE_DIR}/${UPSTART_JOB_NAME}.in")
-
-# build it
-set (pkglibexecdir "${CMAKE_INSTALL_FULL_PKGLIBEXECDIR}")
-configure_file ("${UPSTART_JOB_FILE_IN}" "${UPSTART_JOB_FILE}")
-
-# install it
-install (FILES "${UPSTART_JOB_FILE}"
-         DESTINATION "${UPSTART_JOB_DIR}")
-
-##
 ##  XDG Autostart File
 ##
 
@@ -55,27 +35,6 @@
          DESTINATION "${XDG_AUTOSTART_DIR}")
 
 ##
-##  Upstart XDG Autostart Override
-##
-
-# where to install
-set (UPSTART_XDG_AUTOSTART_DIR "${CMAKE_INSTALL_FULL_DATAROOTDIR}/upstart/xdg/autostart")
-message (STATUS "${UPSTART_XDG_AUTOSTART_DIR} is the Upstart XDG autostart override dir")
-
-set (UPSTART_XDG_AUTOSTART_NAME "${CMAKE_PROJECT_NAME}.upstart.desktop")
-set (UPSTART_XDG_AUTOSTART_FILE "${CMAKE_CURRENT_BINARY_DIR}/${UPSTART_XDG_AUTOSTART_NAME}")
-set (UPSTART_XDG_AUTOSTART_FILE_IN "${CMAKE_CURRENT_SOURCE_DIR}/${UPSTART_XDG_AUTOSTART_NAME}.in")
-
-# build it
-set (pkglibexecdir "${CMAKE_INSTALL_FULL_PKGLIBEXECDIR}")
-configure_file ("${UPSTART_XDG_AUTOSTART_FILE_IN}" "${UPSTART_XDG_AUTOSTART_FILE}")
-
-# install it
-install (FILES "${UPSTART_XDG_AUTOSTART_FILE}"
-         DESTINATION "${UPSTART_XDG_AUTOSTART_DIR}"
-         RENAME "${XDG_AUTOSTART_NAME}")
-
-##
 ##  Unity Indicator File
 ##
 
