diff -Naur indicator-power-0.10/Makefile.in indicator-power/Makefile.in
--- indicator-power-0.10/Makefile.in	2011-10-13 17:16:51.000000000 +0200
+++ indicator-power/Makefile.in	2012-01-11 16:16:10.870851637 +0100
@@ -332,7 +332,7 @@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 SUBDIRS = po
-gsettings_in_file = data/com.canonical.indicator.power.gschema.xml.in
+gsettings_in_file = @srcdir@/data/com.canonical.indicator.power.gschema.xml.in
 gsettings_SCHEMAS = $(gsettings_in_file:.xml.in=.xml)
 
 ###################
