=== modified file 'src/Services/Document.vala'
--- src/Services/Document.vala	2013-09-02 03:46:44 +0000
+++ src/Services/Document.vala	2013-10-02 22:48:25 +0000
@@ -53,6 +53,9 @@
         // Zeitgeist integration
         private ZeitgeistLogger zg_log = new ZeitgeistLogger();
 
+        // It is used to not mark files as changed on load
+        private ulong onchange_handler_id = 0;
+
         // Delegates
         public delegate void VoidFunc ();
 
@@ -115,7 +118,28 @@
             this.working = true;
             message ("Opening \"%s\"", get_basename ());
 
-            
+            onchange_handler_id = this.source_view.buffer.changed.connect (() => {
+                this.source_view.buffer.disconnect(onchange_handler_id);
+                // Signals for SourceView
+                uint timeout_saving = -1;
+                this.source_view.buffer.changed.connect (() => {
+                    check_undoable_actions ();
+                    // Save if autosave is ON
+                    if (settings.autosave) {
+                        if (timeout_saving >= 0) {
+                            Source.remove (timeout_saving);
+                            timeout_saving = -1;
+                        }
+                        timeout_saving = Timeout.add (250, () => {
+                            save ();
+                            timeout_saving = -1;
+                            return false;
+                        });
+                    }
+                    else
+                        this.set_saved_status (false);
+                 });
+            });
 
             // Focus in event for SourceView
             this.source_view.focus_in_event.connect (() => {
@@ -445,25 +469,6 @@
                     this.source_view.set_text (text);
                     this.last_saved_content = text;
                     this.original_content = text;
-                    // Signals for SourceView
-                    uint timeout_saving = -1;
-                    this.source_view.buffer.changed.connect (() => {
-                        check_undoable_actions ();
-                        // Save if autosave is ON
-                        if (settings.autosave) {
-                            if (timeout_saving >= 0) {
-                                Source.remove (timeout_saving);
-                                timeout_saving = -1;
-                            }
-                            timeout_saving = Timeout.add (250, () => {
-                                save ();
-                                timeout_saving = -1;
-                                return false;
-                            });
-                        }
-                        else if (!settings.autosave || file == null)
-                            this.set_saved_status (false);
-                     });
                  });
                  this.loaded = true;
             }

