diff -Naur scratch-9999/plugins/word-completion/completion-provider.vala scratch-9999_patched/plugins/word-completion/completion-provider.vala
--- scratch-9999/plugins/word-completion/completion-provider.vala	2015-02-11 14:40:40.923758000 +0100
+++ scratch-9999_patched/plugins/word-completion/completion-provider.vala	2015-02-11 14:43:58.663485883 +0100
@@ -121,7 +121,7 @@
 
     public bool get_start_iter (Gtk.SourceCompletionContext context,
                                 Gtk.SourceCompletionProposal proposal,
-                                Gtk.TextIter iter) {
+                                out Gtk.TextIter iter) {
         var mark = buffer.get_insert ();
         Gtk.TextIter cursor_iter;
         buffer.get_iter_at_mark (out cursor_iter, mark);
