diff -Naur scratch-2.0.2.orig/CMakeLists.txt scratch-2.0.2/CMakeLists.txt
--- scratch-2.0.2.orig/CMakeLists.txt	2014-02-06 14:40:13.507186592 +0100
+++ scratch-2.0.2/CMakeLists.txt	2014-02-06 14:41:50.694690894 +0100
@@ -44,8 +44,18 @@
 add_definitions(-DVALA_VERSION="${VALA_SHORTVER}")
 add_definitions(-w) # Disable gcc warnings
 
+option (USE_ZEITGEIST "Use Zeitgeist integration" ON)
+
+if (USE_ZEITGEIST)
+  message ("-- Zeitgeist integration enabled")
+  set (ZEITGEIST_DEPS zeitgeist-1.0)
+  set (ZEITGEIST_OPTIONS --define=HAVE_ZEITGEIST)
+else ()
+  message ("-- Zeitgeist integration disabled")
+endif ()
+
 # Dependencies
-set (SCRATCH_DEPS libpeas-1.0 libpeas-gtk-1.0 gtksourceview-3.0 gee-0.8 gobject-2.0 glib-2.0 gio-2.0 gtk+-3.0 granite zeitgeist-1.0)
+set (SCRATCH_DEPS libpeas-1.0 libpeas-gtk-1.0 gtksourceview-3.0 gee-0.8 gobject-2.0 glib-2.0 gio-2.0 gtk+-3.0 granite ${ZEITGEIST_DEPS})
 find_package (PkgConfig)
 pkg_check_modules (GTK+ REQUIRED "gtk+-3.0 >= 3.4")
 pkg_check_modules (LIBSOURCE REQUIRED libpeas-gtk-1.0 gtksourceview-3.0>=3.0 gthread-2.0)
diff -Naur scratch-2.0.2.orig/plugins/brackets-completion/CMakeLists.txt scratch-2.0.2/plugins/brackets-completion/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/brackets-completion/CMakeLists.txt	2014-02-06 14:40:13.507186592 +0100
+++ scratch-2.0.2/plugins/brackets-completion/CMakeLists.txt	2014-02-06 14:43:09.136099794 +0100
@@ -10,7 +10,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --thread
     --vapidir=${CMAKE_BINARY_DIR}
diff -Naur scratch-2.0.2.orig/plugins/browser-preview/CMakeLists.txt scratch-2.0.2/plugins/browser-preview/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/browser-preview/CMakeLists.txt	2014-02-06 14:40:13.507186592 +0100
+++ scratch-2.0.2/plugins/browser-preview/CMakeLists.txt	2014-02-06 14:43:09.136099794 +0100
@@ -17,7 +17,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --thread
     --vapidir=${CMAKE_BINARY_DIR}
diff -Naur scratch-2.0.2.orig/plugins/contractor/CMakeLists.txt scratch-2.0.2/plugins/contractor/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/contractor/CMakeLists.txt	2014-02-06 14:40:13.507186592 +0100
+++ scratch-2.0.2/plugins/contractor/CMakeLists.txt	2014-02-06 14:43:09.136099794 +0100
@@ -11,7 +11,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --vapidir=${CMAKE_BINARY_DIR}
     --vapidir=${CMAKE_BINARY_DIR}/src/
diff -Naur scratch-2.0.2.orig/plugins/filemanager/CMakeLists.txt scratch-2.0.2/plugins/filemanager/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/filemanager/CMakeLists.txt	2014-02-06 14:40:13.507186592 +0100
+++ scratch-2.0.2/plugins/filemanager/CMakeLists.txt	2014-02-06 14:43:09.137099799 +0100
@@ -12,7 +12,7 @@
     scratchcore
     libpeas-1.0
     gtksourceview-3.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --vapidir=${CMAKE_BINARY_DIR}
     --vapidir=${CMAKE_BINARY_DIR}/src/
diff -Naur scratch-2.0.2.orig/plugins/folder-manager/CMakeLists.txt scratch-2.0.2/plugins/folder-manager/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/folder-manager/CMakeLists.txt	2014-02-06 14:40:13.507186592 +0100
+++ scratch-2.0.2/plugins/folder-manager/CMakeLists.txt	2014-02-06 14:43:09.137099799 +0100
@@ -12,7 +12,7 @@
     scratchcore
     libpeas-1.0
     gtksourceview-3.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --vapidir=${CMAKE_BINARY_DIR}
     --vapidir=${CMAKE_BINARY_DIR}/src/
diff -Naur scratch-2.0.2.orig/plugins/open-with/CMakeLists.txt scratch-2.0.2/plugins/open-with/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/open-with/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/open-with/CMakeLists.txt	2014-02-06 14:43:09.137099799 +0100
@@ -9,7 +9,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --vapidir=${CMAKE_BINARY_DIR}
     --vapidir=${CMAKE_BINARY_DIR}/src/
diff -Naur scratch-2.0.2.orig/plugins/outline/CMakeLists.txt scratch-2.0.2/plugins/outline/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/outline/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/outline/CMakeLists.txt	2014-02-06 14:43:09.137099799 +0100
@@ -11,7 +11,7 @@
     scratchcore
     libpeas-1.0
     gtksourceview-3.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 	libvala-${VALA_SHORTVER}
 OPTIONS
     --vapidir=${CMAKE_BINARY_DIR}
diff -Naur scratch-2.0.2.orig/plugins/pastebin/CMakeLists.txt scratch-2.0.2/plugins/pastebin/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/pastebin/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/pastebin/CMakeLists.txt	2014-02-06 14:43:09.137099799 +0100
@@ -13,7 +13,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --vapidir=${CMAKE_BINARY_DIR}
     --vapidir=${CMAKE_BINARY_DIR}/src/
diff -Naur scratch-2.0.2.orig/plugins/source-tree/CMakeLists.txt scratch-2.0.2/plugins/source-tree/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/source-tree/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/source-tree/CMakeLists.txt	2014-02-06 14:43:09.137099799 +0100
@@ -9,7 +9,7 @@
     scratchcore
     libpeas-1.0
     gtksourceview-3.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --vapidir=${CMAKE_BINARY_DIR}
     --vapidir=${CMAKE_BINARY_DIR}/src/
diff -Naur scratch-2.0.2.orig/plugins/spell/CMakeLists.txt scratch-2.0.2/plugins/spell/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/spell/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/spell/CMakeLists.txt	2014-02-06 14:43:09.137099799 +0100
@@ -29,7 +29,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --thread
     --vapidir=${CMAKE_BINARY_DIR}
diff -Naur scratch-2.0.2.orig/plugins/strip-trailing-save/CMakeLists.txt scratch-2.0.2/plugins/strip-trailing-save/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/strip-trailing-save/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/strip-trailing-save/CMakeLists.txt	2014-02-06 14:43:09.137099799 +0100
@@ -10,7 +10,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --thread
     --vapidir=${CMAKE_BINARY_DIR}
diff -Naur scratch-2.0.2.orig/plugins/terminal/CMakeLists.txt scratch-2.0.2/plugins/terminal/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/terminal/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/terminal/CMakeLists.txt	2014-02-06 14:43:09.138099805 +0100
@@ -17,7 +17,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --thread
     --vapidir=${CMAKE_BINARY_DIR}
diff -Naur scratch-2.0.2.orig/plugins/vim-emulation/CMakeLists.txt scratch-2.0.2/plugins/vim-emulation/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/vim-emulation/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/vim-emulation/CMakeLists.txt	2014-02-06 14:43:09.138099805 +0100
@@ -11,7 +11,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --thread
     --vapidir=${CMAKE_BINARY_DIR}
diff -Naur scratch-2.0.2.orig/plugins/word-completion/CMakeLists.txt scratch-2.0.2/plugins/word-completion/CMakeLists.txt
--- scratch-2.0.2.orig/plugins/word-completion/CMakeLists.txt	2014-02-06 14:40:13.506186586 +0100
+++ scratch-2.0.2/plugins/word-completion/CMakeLists.txt	2014-02-06 14:43:09.138099805 +0100
@@ -12,7 +12,7 @@
     scratchcore
     gee-0.8
     libpeas-1.0
-    zeitgeist-1.0
+    ${ZEITGEIST_DEPS}
 OPTIONS
     --thread
     --vapidir=${CMAKE_BINARY_DIR}
diff -Naur scratch-2.0.2.orig/src/CMakeLists.txt scratch-2.0.2/src/CMakeLists.txt
--- scratch-2.0.2.orig/src/CMakeLists.txt	2014-02-06 14:40:13.503186570 +0100
+++ scratch-2.0.2/src/CMakeLists.txt	2014-02-06 14:40:33.443290037 +0100
@@ -24,6 +24,7 @@
 OPTIONS
     --vapidir=${CMAKE_SOURCE_DIR}/vapi
     --thread
+    ${ZEITGEIST_OPTIONS}
 GENERATE_VAPI
     ${LIBNAME}
 GENERATE_HEADER
diff -Naur scratch-2.0.2.orig/src/MainWindow.vala scratch-2.0.2/src/MainWindow.vala
--- scratch-2.0.2.orig/src/MainWindow.vala	2014-02-06 14:40:13.503186570 +0100
+++ scratch-2.0.2/src/MainWindow.vala	2014-02-06 14:40:33.443290037 +0100
@@ -22,7 +22,9 @@
 using Gdk;
 using Pango;
  
+#if HAVE_ZEITGEIST
 using Zeitgeist;
+#endif
 
 using Granite.Widgets;
 using Granite.Services;
@@ -51,9 +53,11 @@
         private Granite.Widgets.ThinPaned hp1;
         private Granite.Widgets.ThinPaned hp2;
         private Granite.Widgets.ThinPaned vp;
-        
+
+#if HAVE_ZEITGEIST
         // Zeitgeist integration
         private Zeitgeist.DataSourceRegistry registry;
+#endif
         
         // Delegates
         delegate void HookFunc ();
@@ -73,6 +77,7 @@
             // Set up layout
             init_layout ();
 
+#if HAVE_ZEITGEIST
             // Set up the Data Source Registry for Zeitgeist
             registry = new DataSourceRegistry ();
 
@@ -92,6 +97,7 @@
                     warning ("%s", reg_err.message);
                 }
             });
+#endif
 
 	    Unix.signal_add (Posix.SIGINT, quit_source_func, Priority.HIGH);
 	    Unix.signal_add (Posix.SIGTERM, quit_source_func, Priority.HIGH);
diff -Naur scratch-2.0.2.orig/src/Services/Document.vala scratch-2.0.2/src/Services/Document.vala
--- scratch-2.0.2.orig/src/Services/Document.vala	2014-02-06 14:40:13.503186570 +0100
+++ scratch-2.0.2/src/Services/Document.vala	2014-02-06 14:40:33.444290042 +0100
@@ -20,7 +20,9 @@
 ***/
 
 using Scratch.Widgets;
+#if HAVE_ZEITGEIST
 using Zeitgeist;
+#endif
 
 namespace Scratch.Services {
 
@@ -50,8 +52,10 @@
         // It is used to load file content on focusing
         private bool loaded = false;
         
+#if HAVE_ZEITGEIST
         // Zeitgeist integration
         private ZeitgeistLogger zg_log = new ZeitgeistLogger();
+#endif
 
         // It is used to not mark files as changed on load
         private ulong onchange_handler_id = 0;
@@ -159,8 +163,10 @@
             // Stop loading
             this.working = false;
 
+#if HAVE_ZEITGEIST
             // Zeitgeist integration
             zg_log.open_insert (file.get_uri (), get_mime_type ());
+#endif
 
             // Grab focus
             this.source_view.grab_focus ();
@@ -219,8 +225,10 @@
             if (file != null) {
                 // Delete backup copy file
                 delete_backup ();
+#if HAVE_ZEITGEIST
                 // Zeitgeist integration
                 zg_log.close_insert (file.get_uri (), get_mime_type ());
+#endif
             }
             
             return ret_value;
@@ -245,8 +253,10 @@
                 warning ("Cannot save \"%s\": %s", get_basename (), e.message);
             }
 
+#if HAVE_ZEITGEIST
             // Zeitgeist integration
             zg_log.save_insert (file.get_uri (), get_mime_type ());
+#endif
 
             doc_saved ();
             this.set_saved_status (true);
@@ -289,8 +299,10 @@
             this.file = new_dest;
             this.save ();
 
+#if HAVE_ZEITGEIST
             // Zeitgeist integration
             zg_log.move_insert (file.get_uri (), new_dest.get_uri (), get_mime_type ());
+#endif
 
             return true;
         }
diff -Naur scratch-2.0.2.orig/src/Services/ZeitgeistLogger.vala scratch-2.0.2/src/Services/ZeitgeistLogger.vala
--- scratch-2.0.2.orig/src/Services/ZeitgeistLogger.vala	2014-02-06 14:40:13.503186570 +0100
+++ scratch-2.0.2/src/Services/ZeitgeistLogger.vala	2014-02-06 14:40:33.444290042 +0100
@@ -18,6 +18,7 @@
   END LICENSE
 ***/
 
+#if HAVE_ZEITGEIST
 using Zeitgeist;
 
 namespace Scratch.Services {
@@ -97,4 +98,5 @@
     }
 
 }
+#endif
 
