diff -Naur /tmp/portage/app-editors/scratch-9999/work/scratch-9999/CMakeLists.txt scratch-9999/CMakeLists.txt
--- /tmp/portage/app-editors/scratch-9999/work/scratch-9999/CMakeLists.txt	2016-08-05 19:34:57.986508979 +0200
+++ scratch-9999/CMakeLists.txt	2016-08-05 19:34:25.287186300 +0200
@@ -106,6 +106,7 @@
 
 
 # Data
+include(Translations)
 configure_file_translation(${CMAKE_SOURCE_DIR}/data/scratch-text-editor.desktop.in ${CMAKE_BINARY_DIR}/data/scratch-text-editor.desktop ${CMAKE_SOURCE_DIR}/po/)
 configure_file_translation(${CMAKE_SOURCE_DIR}/data/scratch-text-editor.appdata.xml.in ${CMAKE_BINARY_DIR}/data/scratch-text-editor.appdata.xml ${CMAKE_SOURCE_DIR}/po/)
 
