=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2013-12-24 13:40:24 +0000
+++ CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -45,7 +45,7 @@
 add_definitions(-w) # Disable gcc warnings
 
 # Dependencies
-set (SCRATCH_DEPS libpeas-1.0 libpeas-gtk-1.0 gtksourceview-3.0 gee-1.0 gobject-2.0 glib-2.0 gio-2.0 gtk+-3.0 granite zeitgeist-1.0)
+set (SCRATCH_DEPS libpeas-1.0 libpeas-gtk-1.0 gtksourceview-3.0 gee-0.8 gobject-2.0 glib-2.0 gio-2.0 gtk+-3.0 granite zeitgeist-1.0)
 find_package (PkgConfig)
 pkg_check_modules (GTK+ REQUIRED "gtk+-3.0 >= 3.4")
 pkg_check_modules (LIBSOURCE REQUIRED libpeas-gtk-1.0 gtksourceview-3.0>=3.0 gthread-2.0)

=== modified file 'plugins/brackets-completion/CMakeLists.txt'
--- plugins/brackets-completion/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/brackets-completion/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -8,7 +8,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/browser-preview/CMakeLists.txt'
--- plugins/browser-preview/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/browser-preview/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -15,7 +15,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/contractor/CMakeLists.txt'
--- plugins/contractor/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/contractor/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -9,7 +9,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/filemanager/CMakeLists.txt'
--- plugins/filemanager/CMakeLists.txt	2013-09-14 09:21:11 +0000
+++ plugins/filemanager/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -7,7 +7,7 @@
     FileView.vala
 PACKAGES
     gtk+-3.0
-    gee-1.0
+    gee-0.8
     granite
     scratchcore
     libpeas-1.0

=== modified file 'plugins/folder-manager/CMakeLists.txt'
--- plugins/folder-manager/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/folder-manager/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -7,7 +7,7 @@
     FileView.vala
 PACKAGES
     gtk+-3.0
-    gee-1.0
+    gee-0.8
     granite
     scratchcore
     libpeas-1.0

=== modified file 'plugins/open-with/CMakeLists.txt'
--- plugins/open-with/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/open-with/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -7,7 +7,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/outline/CMakeLists.txt'
--- plugins/outline/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/outline/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -6,7 +6,7 @@
 	CtagsSymbolResolver.vala
 PACKAGES
     gtk+-3.0
-    gee-1.0
+    gee-0.8
     granite
     scratchcore
     libpeas-1.0

=== modified file 'plugins/pastebin/CMakeLists.txt'
--- plugins/pastebin/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/pastebin/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -11,7 +11,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/source-tree/CMakeLists.txt'
--- plugins/source-tree/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/source-tree/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -4,7 +4,7 @@
 	SourceTreePlugin.vala
 PACKAGES
     gtk+-3.0
-    gee-1.0
+    gee-0.8
     granite
     scratchcore
     libpeas-1.0

=== modified file 'plugins/spell/CMakeLists.txt'
--- plugins/spell/CMakeLists.txt	2013-09-14 09:21:11 +0000
+++ plugins/spell/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -27,7 +27,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/strip-trailing-save/CMakeLists.txt'
--- plugins/strip-trailing-save/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/strip-trailing-save/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -8,7 +8,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/terminal/CMakeLists.txt'
--- plugins/terminal/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/terminal/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -15,7 +15,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/vim-emulation/CMakeLists.txt'
--- plugins/vim-emulation/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/vim-emulation/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -9,7 +9,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'plugins/word-completion/CMakeLists.txt'
--- plugins/word-completion/CMakeLists.txt	2013-08-18 08:13:55 +0000
+++ plugins/word-completion/CMakeLists.txt	2014-01-24 17:15:27 +0000
@@ -10,7 +10,7 @@
     granite
     gtksourceview-3.0
     scratchcore
-    gee-1.0
+    gee-0.8
     libpeas-1.0
     zeitgeist-2.0
 OPTIONS

=== modified file 'src/scratchcore.deps'
--- src/scratchcore.deps	2013-05-26 20:47:33 +0000
+++ src/scratchcore.deps	2014-01-24 17:15:27 +0000
@@ -1,6 +1,6 @@
 scratchcore
 gtksourceview-3.0
-gee-1.0
+gee-0.8
 gobject-2.0
 gio-2.0
 gtk+-3.0

=== modified file 'src/scratchcore.pc.cmake'
--- src/scratchcore.pc.cmake	2013-05-25 07:32:13 +0000
+++ src/scratchcore.pc.cmake	2014-01-24 17:15:27 +0000
@@ -8,5 +8,5 @@
 Version: 0.1  
 Libs: -L@DOLLAR@{libdir} -lscratchcore
 Cflags: -I@DOLLAR@{includedir}/scratch
-Requires: gtk+-3.0 gee-1.0 granite libsoup-2.4
+Requires: gtk+-3.0 gee-0.8 granite libsoup-2.4
 

