--- /src/Views/EthernetPage.vala	2020-04-02 02:10:56.000000000 +0300
+++ /src/Views/EthernetPage.vala	2020-04-20 20:38:48.608134732 +0300
@@ -19,7 +19,7 @@
 
 namespace Network.Widgets {
     public class EtherInterface : Network.Widgets.Page {
-        private Gtk.Revealer top_revealer;
+        private Gtk.Stack widgets_stack;
 
         public EtherInterface (NM.Device device) {
             Object (
@@ -30,20 +30,29 @@
         }
 
         construct {
+            widgets_stack = new Gtk.Stack ();
+            widgets_stack.visible = true;
+
+            var cable_unplugged = new Granite.Widgets.AlertView (_("Cable unplugged"), _("Cable must be connected"), "");
+
             info_box.halign = Gtk.Align.CENTER;
 
-            top_revealer = new Gtk.Revealer ();
+            var top_revealer = new Gtk.Revealer ();
             top_revealer.valign = Gtk.Align.START;
             top_revealer.transition_type = Gtk.RevealerTransitionType.SLIDE_DOWN;
             top_revealer.add (info_box);
 
-            content_area.add (top_revealer);
+            widgets_stack.add_named (cable_unplugged, "unplugged");
+            widgets_stack.add_named (top_revealer, "plugged");
+
+            content_area.add (widgets_stack);
 
             action_area.add (new SettingsButton.from_device (device));
 
             show_all ();
 
             status_switch.bind_property ("active", top_revealer, "reveal-child", GLib.BindingFlags.SYNC_CREATE);
+
             update ();
         }
 
@@ -66,6 +75,21 @@
             base.update ();
 
             state = device.state;
+
+            string? old_visible_name = widgets_stack.get_visible_child_name ();
+            string new_visible_name = "plugged";
+            bool ctrl_switch_sens = true;
+
+            if (state == NM.DeviceState.UNAVAILABLE) {
+                new_visible_name = "unplugged";
+                ctrl_switch_sens = false;
+            }
+
+            status_switch.sensitive = ctrl_switch_sens;
+
+            if (old_visible_name == null || old_visible_name != new_visible_name) {
+                widgets_stack.set_visible_child_name (new_visible_name);
+            }
         }
     }
 }
