=== modified file 'src/ProgressUIHandler.vala'
--- src/ProgressUIHandler.vala	2013-11-21 01:08:15 +0000
+++ src/ProgressUIHandler.vala	2013-11-29 14:01:36 +0000
@@ -64,7 +64,7 @@
     }
 
     private bool server_has_persistence () {
-        unowned List<string> capabilities = Notify.get_server_caps ();
+        List<string> capabilities = Notify.get_server_caps ();
 
         return capabilities.find ("persistence") != null;
     }

