diff -Naur pantheon-files-0.1.5.1.orig/src/ProgressUIHandler.vala pantheon-files-0.1.5.1/src/ProgressUIHandler.vala
--- pantheon-files-0.1.5.1.orig/src/ProgressUIHandler.vala	2014-02-06 20:48:46.427603938 +0100
+++ pantheon-files-0.1.5.1/src/ProgressUIHandler.vala	2014-02-06 20:51:19.334371931 +0100
@@ -64,9 +64,9 @@
     }
 
     private bool server_has_persistence () {
-        unowned List<string> cappabilities = Notify.get_server_caps ();
+        var capabilities = Notify.get_server_caps ();
 
-        return cappabilities.find ("persistence") != null;
+        return capabilities.find ("persistence") != null;
     }
 
     private void progress_info_started_cb (Marlin.Progress.Info info) {
