Index: granite-0.3.0/lib/Widgets/DynamicNotebook.vala
===================================================================
--- granite-0.3.0.orig/lib/Widgets/DynamicNotebook.vala
+++ granite-0.3.0/lib/Widgets/DynamicNotebook.vala
@@ -1,6 +1,6 @@
 /***
     Copyright (C) 2011-2013 Tom Beckmann <tom@elementaryos.org>
-    
+
     This program or library is free software; you can redistribute it
     and/or modify it under the terms of the GNU Lesser General Public
     License as published by the Free Software Foundation; either
@@ -658,10 +658,19 @@ namespace Granite.Widgets {
        /**
         * The text shown in the add button tooltip
         */
+#if VALA_0_26
+        public string add_button_tooltip {
+            get { _add_button_tooltip = add_button.tooltip_text; return _add_button_tooltip; }
+            set { add_button.tooltip_text = value; }
+        }
+        // Use temporary field to avoid breaking API this can be dropped while preparing for 0.4
+        string _add_button_tooltip;
+#else
         public string add_button_tooltip {
             get { return add_button.tooltip_text; }
             set { add_button.tooltip_text = value; }
         }
+#endif
 
         public Tab current {
             get { return tabs.nth_data (notebook.get_current_page ()); }
