diff -Naur gsettings-desktop-schemas-3.8.0/schemas/org.gnome.desktop.background.gschema.xml.in.in gsettings-desktop-schemas-3.8.0.patched/schemas/org.gnome.desktop.background.gschema.xml.in.in
--- gsettings-desktop-schemas-3.8.0/schemas/org.gnome.desktop.background.gschema.xml.in.in	2013-02-21 12:19:18.000000000 +0100
+++ gsettings-desktop-schemas-3.8.0.patched/schemas/org.gnome.desktop.background.gschema.xml.in.in	2013-05-14 14:04:24.236419618 +0200
@@ -1,5 +1,10 @@
 <schemalist gettext-domain="gsettings-desktop-schemas">
   <schema id="org.gnome.desktop.background" path="/org/gnome/desktop/background/">
+    <key name="draw-background" type="b">
+      <default>true</default>
+      <_summary>Draw Desktop Background</_summary>
+      <_description>Have GNOME draw the desktop background.</_description>
+    </key>
     <key name="picture-options" enum="org.gnome.desktop.GDesktopBackgroundStyle">
       <default>'zoom'</default>
       <_summary>Picture Options</_summary>
