=== modified file 'debian/pantheon.session'
--- debian/pantheon.session	2013-10-01 19:45:05 +0000
+++ debian/pantheon.session	2013-11-17 12:46:31 +0000
@@ -1,7 +1,5 @@
 [GNOME Session]
 Name=Pantheon
-RequiredComponents=gnome-settings-daemon;
-RequiredProviders=windowmanager;
-DefaultProvider-windowmanager=gala
+RequiredComponents=gala;gnome-settings-daemon;
 FallbackSession=ubuntu
 DesktopName=Pantheon

