Author: Romain Perier <mrpouet@gentoo.org>
Subject: Missing gconf flags for gnome plugin, fix bug #282807
Date: 2009-08-29 12:25

---
 gtk/gnome/Makefile.am |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/gtk/gnome/Makefile.am b/gtk/gnome/Makefile.am
index 208318a..d0a7c54 100644
--- a/gtk/gnome/Makefile.am
+++ b/gtk/gnome/Makefile.am
@@ -2,7 +2,7 @@ moduledir = $(DESTDIR)$(windowsettingslibdir)/window-manager-settings
 
 if USE_GNOME
 libcompiz_la_LDFLAGS = -export-dynamic -avoid-version -rpath $(moduledir)
-libcompiz_la_LIBADD  = @GNOME_WINDOW_SETTINGS_LIBS@
+libcompiz_la_LIBADD  = @GNOME_WINDOW_SETTINGS_LIBS@ @GCONF_LIBS@
 libcompiz_la_SOURCES = 		\
 	compiz-window-manager.c	\
 	compiz-window-manager.h
@@ -76,8 +76,9 @@ if USE_GNOME_KEYBINDINGS
 endif
 endif
 
-INCLUDES = @GNOME_WINDOW_SETTINGS_CFLAGS@			   \
-	-DMETACITY_THEME_DIR=\""$(windowsettingsdatadir)/themes"\" \
+INCLUDES = @GNOME_WINDOW_SETTINGS_CFLAGS@				\
+	@GCONF_CFLAGS@							\
+	-DMETACITY_THEME_DIR=\""$(windowsettingsdatadir)/themes"\" 	\
 	-DLOCALEDIR="\"@datadir@/locale"\"
 
 noinst_LTLIBRARIES = \
