--- compiz-0.8.4/gtk/window-decorator/gtk-window-decorator.c.old	2010-03-28 13:01:58.639940630 +0200
+++ compiz-0.8.4/gtk/window-decorator/gtk-window-decorator.c	2010-03-28 13:02:30.580354171 +0200
@@ -4233,7 +4233,7 @@
 static void
 hide_tooltip (void)
 {
-    if (GTK_WIDGET_VISIBLE (tip_window))
+    if (gtk_widget_get_visible (tip_window))
 	g_get_current_time (&tooltip_last_popdown);
 
     gtk_widget_hide (tip_window);
