diff -Naur compiz-0.9.5.92.orig/cmake/CompizGSettings.cmake compiz-0.9.5.92/cmake/CompizGSettings.cmake
--- compiz-0.9.5.92.orig/cmake/CompizGSettings.cmake	2011-08-26 09:18:18.378126233 +0200
+++ compiz-0.9.5.92/cmake/CompizGSettings.cmake	2011-08-26 09:19:15.074122072 +0200
@@ -35,8 +35,6 @@
 		    file (INSTALL DESTINATION \"${GSETTINGS_GLOBAL_INSTALL_DIR}\"
 			  TYPE FILE
 			  FILES \"${_src}\")
-		    message (\"-- Recompiling GSettings schemas in ${GSETTINGS_GLOBAL_INSTALL_DIR}\"\)
-		    execute_process (COMMAND ${GLIB_COMPILE_SCHEMAS} ${GSETTINGS_GLOBAL_INSTALL_DIR})
 
 		else (\"$ENV{USER}\"\ STREQUAL \"root\"\)
 		    # It seems like this is only available in CMake > 2.8.5
@@ -49,8 +47,6 @@
 		    file (INSTALL DESTINATION \"${COMPIZ_DESTDIR}${_dst}\"
 			  TYPE FILE
 			  FILES \"${_src}\")
-		    message (\"-- Recompiling GSettings schemas in ${COMPIZ_DESTDIR}${_dst}\"\)
-		    execute_process (COMMAND ${GLIB_COMPILE_SCHEMAS} ${COMPIZ_DESTDIR}${_dst})
 		endif (\"$ENV{USER}\" STREQUAL \"root\"\)
 		")
     endif ()
