diff -p -up compiz-0.7.8/configure.ac.framesvg compiz-0.7.8/configure.ac
--- compiz-0.7.8/configure.ac.framesvg	2008-11-20 15:58:43.000000000 +0100
+++ compiz-0.7.8/configure.ac	2008-11-20 16:34:25.000000000 +0100
@@ -487,7 +487,7 @@ if test "x$use_kde4" = "xyes"; then
   kde4libs=`kde4-config --install lib --expandvars 2>/dev/null`
   kde4incs=`kde4-config --install include --expandvars 2>/dev/null`
 
-  kde4_test_includes="kdecoration.h kcommondecoration.h kdecorationbridge.h plasma/panelsvg.h"
+  kde4_test_includes="kdecoration.h kcommondecoration.h kdecorationbridge.h plasma/framesvg.h"
 
   dnl Check for KWin headers
   AC_MSG_CHECKING([for KWin headers])
diff -p -up compiz-0.7.8/kde/window-decorator-kde4/switcher.cpp.framesvg compiz-0.7.8/kde/window-decorator-kde4/switcher.cpp
--- compiz-0.7.8/kde/window-decorator-kde4/switcher.cpp.framesvg	2008-08-08 13:20:31.000000000 +0200
+++ compiz-0.7.8/kde/window-decorator-kde4/switcher.cpp	2008-11-20 16:38:10.000000000 +0100
@@ -30,7 +30,7 @@
 
 #include <fixx11h.h>
 
-#include <KDE/Plasma/PanelSvg>
+#include <KDE/Plasma/FrameSvg>
 #include <KDE/Plasma/Theme>
 
 #include <kwindowsystem.h>
@@ -46,9 +46,9 @@ mId (id)
     QColor   color;
     color = Plasma::Theme::defaultTheme ()->color (Plasma::Theme::TextColor);
 
-    mBackground = new Plasma::PanelSvg();
+    mBackground = new Plasma::FrameSvg();
     mBackground->setImagePath ("dialogs/background");
-    mBackground->setEnabledBorders(Plasma::PanelSvg::AllBorders);
+    mBackground->setEnabledBorders(Plasma::FrameSvg::AllBorders);
 
     mBorder.left   = mBackground->marginSize(Plasma::LeftMargin);
     mBorder.right  = mBackground->marginSize(Plasma::RightMargin);
@@ -131,8 +131,8 @@ KWD::Switcher::redrawPixmap ()
     p.setCompositionMode (QPainter::CompositionMode_Source);
     p.setRenderHint (QPainter::SmoothPixmapTransform);
 
-    mBackground->resizePanel (QSizeF (contentWidth, contentHeight));
-    mBackground->paintPanel (&p, QRect (0, 0, contentWidth,
+    mBackground->resizeFrame (QSizeF (contentWidth, contentHeight));
+    mBackground->paintFrame (&p, QRect (0, 0, contentWidth,
 			contentHeight));
 
     mBackgroundPixmap = mPixmap.copy (mBorder.left, mBorder.top,
@@ -168,7 +168,7 @@ KWD::Switcher::update ()
     p.setCompositionMode (QPainter::CompositionMode_Source);
 
 
-    mBackground->paintPanel (&p, QRect (mBorder.left, mBorder.top +
+    mBackground->paintFrame (&p, QRect (mBorder.left, mBorder.top +
 			mGeometry.height () + 5, mGeometry.width (),
 			fm.height ()));
 
diff -p -up compiz-0.7.8/kde/window-decorator-kde4/switcher.h.framesvg compiz-0.7.8/kde/window-decorator-kde4/switcher.h
--- compiz-0.7.8/kde/window-decorator-kde4/switcher.h.framesvg	2008-08-08 13:20:31.000000000 +0200
+++ compiz-0.7.8/kde/window-decorator-kde4/switcher.h	2008-11-20 16:36:22.000000000 +0100
@@ -30,7 +30,7 @@
 
 namespace Plasma
 {
-class PanelSvg;
+class FrameSvg;
 }
 
 class QSpacerItem;
@@ -70,7 +70,7 @@ class Switcher
 
 	QRect mGeometry;
 
-	Plasma::PanelSvg *mBackground;
+	Plasma::FrameSvg *mBackground;
 	QPixmap mPixmap;
 	QPixmap mBackgroundPixmap;
 
