From 0f95c41a0aa175ddf7947ba18b01f746c95594a9 Mon Sep 17 00:00:00 2001
From: Paul Donohue <compiz@paulsd.com>
Date: Sun, 17 Oct 2010 16:24:14 +0000
Subject: Fix pixmap size calculation, server drawn borders are also included in 
the pixmap.
---
diff --git a/src/window.c b/src/window.c
index a925288..6674643 100644
--- a/src/window.c
+++ b/src/window.c
@@ -1446,8 +1446,8 @@ bindWindow (CompWindow *w)
 	}
 
 	w->pixmap = XCompositeNameWindowPixmap (dpy, w->id);
-	w->width  = attr.width;
-	w->height = attr.height;
+	w->width  = attr.width + attr.border_width * 2;
+	w->height = attr.height + attr.border_width * 2;
 
 	XUngrabServer (dpy);
     }
--
cgit v0.8.3.1-30-gff3a
