=== added directory 'openbox-3'
=== added file 'openbox-3/bullet.xbm'
--- openbox-3/bullet.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/bullet.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define bullet_width 4
+#define bullet_height 7
+static unsigned char bullet_bits[] = {
+   0x01, 0x03, 0x07, 0x0f, 0x07, 0x03, 0x01 };

=== added file 'openbox-3/close.xbm'
--- openbox-3/close.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/close.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define close_width 6
+#define close_height 6
+static unsigned char close_bits[] = {
+   0x33, 0x3f, 0x1e, 0x1e, 0x3f, 0x33 };

=== added file 'openbox-3/close_hover.xbm'
--- openbox-3/close_hover.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/close_hover.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define close_width 6
+#define close_height 6
+static unsigned char close_bits[] = {
+   0x33, 0x3f, 0x1e, 0x1e, 0x3f, 0x33 };

=== added file 'openbox-3/close_pressed.xbm'
--- openbox-3/close_pressed.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/close_pressed.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define close_width 6
+#define close_height 6
+static unsigned char close_bits[] = {
+   0x33, 0x3f, 0x1e, 0x1e, 0x3f, 0x33 };

=== added file 'openbox-3/desk.xbm'
--- openbox-3/desk.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/desk.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define desk_width 6
+#define desk_height 6
+static unsigned char desk_bits[] = {
+   0x33, 0x33, 0x00, 0x00, 0x33, 0x33 };

=== added file 'openbox-3/desk_toggle.xbm'
--- openbox-3/desk_toggle.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/desk_toggle.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define desk_toggle_width 6
+#define desk_toggle_height 6
+static unsigned char desk_toggle_bits[] = {
+   0x00, 0x1e, 0x1a, 0x16, 0x1e, 0x00 };

=== added file 'openbox-3/iconify.xbm'
--- openbox-3/iconify.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/iconify.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define iconify_width 6
+#define iconify_height 6
+static unsigned char iconify_bits[] = {
+   0x00, 0x00, 0x3f, 0x3f, 0x00, 0x00 };

=== added file 'openbox-3/iconify_hover.xbm'
--- openbox-3/iconify_hover.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/iconify_hover.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define iconify_width 6
+#define iconify_height 6
+static unsigned char iconify_bits[] = {
+   0x00, 0x00, 0x3f, 0x3f, 0x00, 0x00 };

=== added file 'openbox-3/iconify_pressed.xbm'
--- openbox-3/iconify_pressed.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/iconify_pressed.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define iconify_width 6
+#define iconify_height 6
+static unsigned char iconify_bits[] = {
+   0x00, 0x00, 0x3f, 0x3f, 0x00, 0x00 };

=== added file 'openbox-3/max.xbm'
--- openbox-3/max.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/max.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define max_width 6
+#define max_height 6
+static unsigned char max_bits[] = {
+   0x0c, 0x0c, 0x3f, 0x3f, 0x0c, 0x0c };

=== added file 'openbox-3/max_disabled.xbm'
--- openbox-3/max_disabled.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/max_disabled.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define max_width 6
+#define max_height 6
+static unsigned char max_bits[] = {
+   0x0c, 0x0c, 0x3f, 0x3f, 0x0c, 0x0c };

=== added file 'openbox-3/max_hover.xbm'
--- openbox-3/max_hover.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/max_hover.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define max_width 6
+#define max_height 6
+static unsigned char max_bits[] = {
+   0x0c, 0x0c, 0x3f, 0x3f, 0x0c, 0x0c };

=== added file 'openbox-3/max_toggled.xbm'
--- openbox-3/max_toggled.xbm	1970-01-01 00:00:00 +0000
+++ openbox-3/max_toggled.xbm	2011-06-17 04:02:26 +0000
@@ -0,0 +1,4 @@
+#define max_width 6
+#define max_height 6
+static unsigned char max_bits[] = {
+   0x0c, 0x0c, 0x3f, 0x3f, 0x0c, 0x0c };

=== added file 'openbox-3/themerc'
--- openbox-3/themerc	1970-01-01 00:00:00 +0000
+++ openbox-3/themerc	2011-06-17 04:02:26 +0000
@@ -0,0 +1,105 @@
+# Author: Cody Garver <codygarver@gmail.com>
+# Author of original metacity theme: elementary Art Team
+# elementary theme for openbox.
+# License: GPL
+# Date: June 2011
+
+# menu settings
+menu.items.active.bg: Gradient Vertical Flat Border
+menu.items.active.bg.color: #B6CCE0
+menu.items.active.bg.colorTo: #82A3BF
+menu.items.active.bg.border.color: #666666
+menu.items.active.text.color: #FFFFFF
+menu.items.bg: Solid Flat
+menu.items.bg.color: #383838
+menu.items.disabled.text.color: #999999
+menu.items.font: shadow=n
+menu.items.text.color: #FFFFFF
+menu.overlap.x: 1
+menu.separator.color: #444444
+menu.separator.padding.width: 10
+menu.title.bg: Gradient Vertical Flat Border
+menu.title.bg.color: #B6CCE0
+menu.title.bg.colorTo: #82A3BF
+menu.title.bg.border.color: #666666
+menu.title.text.color: #FFFFFF
+menu.title.text.justify: center
+menu.title.text.font: shadow=y:shadowtint=100:shadowoffset=1
+
+# osd settings
+osd.bg: Solid Raised
+osd.bg.color: #383838
+osd.bg.border.color: #151515
+osd.hilight.bg: Solid Flat
+osd.hilight.bg.color: #666666
+osd.label.bg: Gradient Vertical Flat
+osd.label.bg.color: #999999
+osd.label.bg.colorTo: #666666
+osd.label.text.color: #666666
+
+
+# window settings (focused)
+window.active.button.disabled.bg: ParentRelative Flat
+window.active.button.disabled.image.color: #A39E9E
+window.active.button.hover.bg: ParentRelative Flat
+window.active.button.hover.image.color: #151515
+window.active.button.pressed.bg: parentrelative
+window.active.button.pressed.bg.color: #3c3c3c
+window.active.button.pressed.bg.colorTo: #3c3c3c
+window.active.button.pressed.image.color: #000000
+window.active.button.toggled.hover.image.color: #151515
+window.active.button.toggled.image.color: #151515
+window.active.button.toggled.pressed.image.color: #3C3C3C
+window.active.button.toggled.unpressed.image.color: #3C3C3C
+window.active.button.unpressed.bg: parentrelative
+window.active.button.unpressed.bg.color: #3c3c3c
+window.active.button.unpressed.bg.colorTo: #3c3c3c
+window.active.button.unpressed.image.color: #696969
+window.active.client.color: #616161
+window.active.grip.bg: Solid Flat
+window.active.grip.bg.color: #D6D6D6
+window.active.handle.bg: Solid Flat
+window.active.handle.bg.color: #D5D5D5
+window.active.label.bg: parentrelative
+window.active.label.bg.color: #3c3c3c
+window.active.label.bg.colorTo: #3c3c3c
+window.active.label.text.color: #6F6F6F
+window.active.label.text.font: shadow=n
+window.active.title.bg: Gradient Vertical Flat
+window.active.title.bg.color: #EDEDED
+window.active.title.bg.colorTo: #DEDEDE
+window.active.title.bg.shadow: 0
+
+
+
+# window settings (unfocused)
+window.inactive.button.disabled.bg: ParentRelative Flat
+window.inactive.button.disabled.image.color: #3C3C3C
+window.inactive.button.pressed.bg: ParentRelative Flat
+window.inactive.button.toggled.image.color: #616161
+window.inactive.button.toggled.pressed.image.color: #545454
+window.inactive.button.toggled.unpressed.image.color: #3C3C3C
+window.inactive.button.unpressed.bg: parentrelative
+window.inactive.button.unpressed.image.color: #696969
+window.inactive.client.color: #3C3C3C
+window.inactive.grip.bg: Gradient Vertical Flat
+window.inactive.grip.bg.color: #555555
+window.inactive.grip.bg.colorTo: #3C3C3C
+window.inactive.handle.bg: Gradient Vertical Flat
+window.inactive.handle.bg.color: #575757
+window.inactive.handle.bg.colorTo: #3C3C3C
+window.inactive.label.bg: ParentRelative Flat
+window.inactive.label.text.color: #999999
+window.inactive.label.text.font: shadow=n
+window.inactive.title.bg: Gradient Vertical Flat
+window.inactive.title.bg.color: #EDEDED
+window.inactive.title.bg.colorTo: #DEDEDE
+
+# everything else
+window.label.text.justify: Center
+padding.height: 3
+padding.width: 5
+border.color: #DEDEDE
+border.width: 1
+window.client.padding.width: 0
+window.handle.width: 2

