From fbc9ab35ad31371301df72fd194e723a33ea7d92 Mon Sep 17 00:00:00 2001
From: lenemter <lenemter@gmail.com>
Date: Tue, 20 Aug 2024 22:48:24 +0900
Subject: [PATCH] Fix mutter 45

---
 vapi/libmutter.vapi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vapi/libmutter.vapi b/vapi/libmutter.vapi
index 3bcc051ed..23373066e 100644
--- a/vapi/libmutter.vapi
+++ b/vapi/libmutter.vapi
@@ -792,7 +792,11 @@ namespace Meta {
 	[CCode (cheader_filename = "meta/meta-selection-source-memory.h", type_id = "meta_selection_source_memory_get_type ()")]
 	public sealed class SelectionSourceMemory : Meta.SelectionSource {
 		[CCode (has_construct_function = false, type = "MetaSelectionSource*")]
+#if HAS_MUTTER46
 		public SelectionSourceMemory (string mimetype, GLib.Bytes content) throws GLib.Error;
+#else
+		public SelectionSourceMemory (string mimetype, GLib.Bytes content);
+#endif
 	}
 	[CCode (cheader_filename = "meta/meta-settings.h", has_type_id = false)]
 	[Compact]
