From 5216e59352924c89326fbdaba42e2662565a8fe9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20For=C3=A9?= <daniel@elementary.io>
Date: Thu, 9 Jul 2020 12:30:11 -0700
Subject: [PATCH] BackgroundSource: use map values, not map entries

---
 src/Background/BackgroundSource.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Background/BackgroundSource.vala b/src/Background/BackgroundSource.vala
index bf9353cf..a84c3e08 100644
--- a/src/Background/BackgroundSource.vala
+++ b/src/Background/BackgroundSource.vala
@@ -119,7 +119,7 @@ namespace Gala {
             screen.monitors_changed.disconnect (monitors_changed);
 #endif
 
-            foreach (var background in backgrounds) {
+            foreach (var background in backgrounds.values) {
                 background.changed.disconnect (background_changed);
                 background.destroy ();
             }
