=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2012-05-03 06:59:37 +0000
+++ CMakeLists.txt	2014-01-27 21:41:08 +0000
@@ -63,14 +63,14 @@
 # Lingo
 #
 
-set (LINGO_DEPS "json-glib-1.0;gtk+-3.0;sqlite3;libsoup-2.4;gee-1.0;granite;")
+set (LINGO_DEPS "json-glib-1.0;gtk+-3.0;sqlite3;libsoup-2.4;gee-0.8;granite;")
 
 find_package(PkgConfig)
 pkg_check_modules(JSON REQUIRED "json-glib-1.0")
 pkg_check_modules(SQLITE3 REQUIRED "sqlite3")
 pkg_check_modules(SOUP REQUIRED "libsoup-2.4")
 pkg_check_modules(GTK+ REQUIRED "gtk+-3.0")
-pkg_check_modules(GEE REQUIRED "gee-1.0")
+pkg_check_modules(GEE REQUIRED "gee-0.8")
 #pkg_check_modules(UNIQUE REQUIRED "unique-1.0")
 pkg_check_modules(DEPS REQUIRED ${LINGO_DEPS} gthread-2.0)
 
@@ -97,7 +97,7 @@
 	gtk+-3.0
     granite
 	json-glib-1.0
-	gee-1.0
+	gee-0.8
 	libsoup-2.4
 	sqlite3
 CUSTOM_VAPIS

