=== modified file 'icons/CMakeLists.txt'
--- icons/CMakeLists.txt	2011-08-08 16:57:40 +0000
+++ icons/CMakeLists.txt	2012-05-18 18:22:51 +0000
@@ -1,6 +1,8 @@
 install(FILES thumbnail_frame.png DESTINATION share/pixmaps/marlin/)
 install(FILES marlin.svg view-list-column-symbolic.svg view-list-compact-symbolic.svg view-list-details-symbolic.svg view-list-icons-symbolic.svg DESTINATION share/icons/hicolor/scalable/apps/)
-install (CODE "message(\"-- Update Gtk icon cache...\")
-")
-install (CODE "execute_process(COMMAND gtk-update-icon-cache -f -t ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor)
-")
+
+option (ICON_UPDATE "Run gtk-update-icon-cache after the install." ON)
+if (ICON_UPDATE)
+  install (CODE "message(\"-- Updating icon cache...\")")
+  install (CODE "execute_process(COMMAND gtk-update-icon-cache -f -t ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor)")
+endif ()

