=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2013-01-21 15:52:36 +0000
+++ CMakeLists.txt	2013-03-26 09:35:26 +0000
@@ -41,39 +41,47 @@
 
 find_package (PkgConfig)
 
-# Alternative Gtk+ support
-pkg_check_modules (GTK gtk+-3.0)
+set (PKG_DEPS
+    sqlheavy-0.1
+    gee-1.0)
+
+set (PKG_DEPS_CHECK
+    packagekit-glib2
+    sqlheavy-0.1
+    gthread-2.0
+    gee-1.0)
+set (VALAC_OPTIONS
+     --thread
+     --target-glib=2.32)
+
+# Optional Gtk+ support
+option (DISABLE_GTK "Disable Gtk+ support auto-detection" OFF)
+option (WANT_GTK "Enable Gtk+ support" OFF)
+
+set (GTK_DEPS gtk+-3.0)
+set (GTK_DEPS_CHECK gtk+-3.0>=3.3.14)
+
+if(WANT_GTK)
+  message(STATUS "Enabled Gtk+ support.")
+  pkg_check_modules (GTK REQUIRED ${GTK_DEPS_CHECK})
+else()
+  if(DISABLE_GTK)
+    message(STATUS "Disabled Gtk+ auto-detection (Gtk+ support will be disabled).")
+  else ()
+    message(STATUS "Using Gtk+ auto-detection (Gtk+ support might be enabled).")
+    pkg_check_modules (GTK ${GTK_DEPS_CHECK})
+    if (${GTK_FOUND})
+      message("-- Gtk+ support, ON")
+    else ()
+      message("-- Gtk+ support, OFF")
+    endif ()
+  endif()
+endif()
+
 if (${GTK_FOUND})
-    set (PKG_DEPS
-        gtk+-3.0
-        sqlheavy-0.1
-        gee-1.0)
-
-    set (PKG_DEPS_CHECK
-        gtk+-3.0>=3.3.14
-        packagekit-glib2
-        sqlheavy-0.1
-        gthread-2.0
-        gee-1.0)
-    set (VALAC_OPTIONS
-     --thread
-     --target-glib=2.32
-     -D HAVE_GTK)
-    message("-- Gtk+ support, ON")
-else ()
-    set (PKG_DEPS
-        sqlheavy-0.1
-        gee-1.0)
-
-    set (PKG_DEPS_CHECK
-        packagekit-glib2
-        sqlheavy-0.1
-        gthread-2.0
-        gee-1.0)
-    set (VALAC_OPTIONS
-     --thread
-     --target-glib=2.32)
-    message("-- Gtk+ support, OFF")
+    set (PKG_DEPS ${GTK_DEPS} ${PKG_DEPS})
+    set (PKG_DEPS_CHECK ${GTK_DEPS_CHECK} ${PKG_DEPS_CHECK})
+    set (VALAC_OPTIONS ${VALAC_OPTIONS} -D HAVE_GTK)
 endif ()
 
 # GI dependencies

