=== modified file 'src/contacts-linking.vala'
--- src/contacts-linking.vala	2012-02-13 16:10:14 +0000
+++ src/contacts-linking.vala	2012-03-11 12:52:33 +0000
@@ -21,7 +21,7 @@
 using Gee;
 
 namespace Contacts {
-  internal abstract class PersonaAttribute : Object {
+  public abstract class PersonaAttribute : Object {
     public string property_name;
 
     public static HashSet<PersonaAttribute> create_set () {

