=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2011-12-03 15:06:51 +0000
+++ CMakeLists.txt	2012-05-18 14:17:06 +0000
@@ -14,4 +14,4 @@
 
 # GSettings schema
 include(GSettings)
-add_schema("org.elementary.cerbere.gschema.xml")
+add_schema("org.pantheon.cerbere.gschema.xml")

=== renamed file 'org.elementary.cerbere.gschema.xml' => 'org.pantheon.cerbere.gschema.xml'
--- org.elementary.cerbere.gschema.xml	2012-03-29 00:51:50 +0000
+++ org.pantheon.cerbere.gschema.xml	2012-05-18 14:17:06 +0000
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <schemalist>
-    <schema path="/apps/cerbere/settings/" id="org.elementary.cerbere.settings" gettext-domain="cerbere">
+    <schema path="/org/pantheon/cerbere/settings/" id="org.pantheon.cerbere.settings" gettext-domain="cerbere">
         <key type="as" name="desktop-components">
             <default>['wingpanel','plank','/usr/lib/notify-osd/notify-osd']</default>
             <summary>List of desktop components</summary>

=== modified file 'src/Cerbere.vala'
--- src/Cerbere.vala	2012-03-09 05:15:32 +0000
+++ src/Cerbere.vala	2012-05-18 14:17:06 +0000
@@ -37,7 +37,7 @@
         private string[] desktop_bins;
 
         construct {
-            application_id = "org.elementary.cerbere";
+            application_id = "org.pantheon.cerbere";
             flags = GLib.ApplicationFlags.IS_SERVICE;
         }
 
@@ -54,7 +54,7 @@
         }
 
         void load_config () {
-            var settings = new Settings ("org.elementary.cerbere.settings");
+            var settings = new Settings ("org.pantheon.cerbere.settings");
             desktop_bins = settings.get_strv ("desktop-components");
             max_crashes = settings.get_int ("max-crashes");
             crash_time = settings.get_double ("crash-time");

