--- /src/Cerbere.vala	2019-07-11 17:21:02.000000000 +0300
+++ /src/Cerbere.vala	2019-08-02 23:03:12.779795700 +0300
@@ -40,7 +40,9 @@
         base.startup ();
 
         // Try to register Cerbere with the session manager.
-        register_session_client ();
+        if (!register_session_client ()) {
+            return;
+        }
 
         settings = new SettingsManager ();
         start_processes (settings.process_list);
@@ -52,25 +54,23 @@
         hold ();
     }
 
-    private void register_session_client () {
+    private bool register_session_client () {
         if (sm_client != null)
-            return;
+            return false;
 
         sm_client = new SessionManager.Client (application_id);
 
         try {
             sm_client.register ();
-        } catch (SessionManager.ConnectionError e) {
-            critical (e.message);
-            return_if_reached ();
-        }
-
-        if (sm_client != null) {
             // The session manager may ask us to quit the service, and so we do.
             sm_client.stop_service.connect (quit_service);
             // Cleanly shutdown when receiving SIGTERM or SIGHUP.
             Posix.signal (Posix.Signal.TERM, handle_sigterm);
             Posix.signal (Posix.Signal.HUP, handle_sighup);
+            return true;
+        } catch (SessionManager.ConnectionError e) {
+            critical (e.message);
+            return false;
         }
     }
 
