From 6ca78b3d3890786d11b5fdb934260f9082e4561a Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Tue, 11 Jul 2023 12:55:35 +0200
Subject: [PATCH] lib: Wnck.set_client_type() needs to be called before using
 Wnck

---
 lib/Services/WindowControl.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Services/WindowControl.vala b/lib/Services/WindowControl.vala
index e3876126..07f1022f 100644
--- a/lib/Services/WindowControl.vala
+++ b/lib/Services/WindowControl.vala
@@ -52,10 +52,10 @@ namespace Plank
 		
 		public static void initialize ()
 		{
-			unowned Wnck.Screen screen = Wnck.Screen.get_default ();
-			
 			Wnck.set_client_type (Wnck.ClientType.PAGER);
 			
+			unowned Wnck.Screen screen = Wnck.Screen.get_default ();
+			
 			// Make sure internal window-list of Wnck is most up to date
 			Gdk.error_trap_push ();
 			screen.force_update ();
