diff -Naur geary-0.2.2/src/client/ui/main-window.vala geary-0.2.2-patched/src/client/ui/main-window.vala
--- geary-0.2.2/src/client/ui/main-window.vala	2012-10-19 00:44:04.000000000 +0200
+++ geary-0.2.2-patched/src/client/ui/main-window.vala	2012-11-29 22:58:48.155841657 +0100
@@ -19,8 +19,8 @@
     private bool window_maximized;
 
 #if HAVE_LIBGRANITE
-    private Granite.Widgets.SidebarPaned folder_paned = new Granite.Widgets.SidebarPaned();
-    private Granite.Widgets.SidebarPaned conversations_paned = new Granite.Widgets.SidebarPaned();
+    private Granite.Widgets.ThinPaned folder_paned = new Granite.Widgets.ThinPaned();
+    private Granite.Widgets.ThinPaned conversations_paned = new Granite.Widgets.ThinPaned();
 #else
     private Gtk.Paned folder_paned = new Gtk.Paned(Gtk.Orientation.HORIZONTAL);
     private Gtk.Paned conversations_paned = new Gtk.Paned(Gtk.Orientation.HORIZONTAL);
