From: Julian Ospald <hasufell@gentoo.org>
Date: Thu May 30 11:56:51 UTC 2013
Subject: respect system cflags

--- geary-0.3.0/src/CMakeLists.txt
+++ geary-0.3.0/src/CMakeLists.txt
@@ -407,16 +407,8 @@
     -D_GSETTINGS_DIR=\"${CMAKE_BINARY_DIR}/gsettings\"
     -DGETTEXT_PACKAGE=\"${GETTEXT_PACKAGE}\"
     -DLANGUAGE_SUPPORT_DIRECTORY=\"${LANGUAGE_SUPPORT_DIRECTORY}\"
-    -g
 )
 
-if (NOT DEBUG)
-    set(CFLAGS
-        ${CFLAGS}
-        -O2
-    )    
-endif ()
-
 if (REF_TRACKING)
     message(STATUS "Reference tracking: ON")
     set(EXTRA_VALA_OPTIONS
