diff -Naur indicator-network-0.3.8/src/indicator/Makefile.am indicator-network-0.3.8.patched/src/indicator/Makefile.am
--- indicator-network-0.3.8/src/indicator/Makefile.am	2011-03-09 14:27:41.000000000 +0100
+++ indicator-network-0.3.8.patched/src/indicator/Makefile.am	2014-03-12 13:40:15.641772271 +0100
@@ -39,8 +39,7 @@
         --pkg=gdk-3.0 \
         --pkg=gee-1.0 \
         --pkg=pango \
-        --pkg=cairo \
-	--fatal-warnings
+        --pkg=cairo
 
 toggleswitch_APIFILES = \
   toggleswitch.h 
--- indicator-network-0.3.8.orig/src/settings/Makefile.am	2014-03-12 14:10:42.515574951 +0100
+++ indicator-network-0.3.8/src/settings/Makefile.am	2014-03-12 14:11:03.267653328 +0100
@@ -35,8 +35,7 @@
         --pkg=pango \
         --pkg=cairo \
 	--pkg=config \
-	--pkg=Connman-1.0 \
-	--fatal-warnings
+	--pkg=Connman-1.0
 
 # -w is needed to disable gcc warnings because of valac
 indicator_network_settings_CFLAGS = \
