diff -urpN libindicator-12.10.2+14.04.20141007.orig/libindicator/indicator3-0.4.pc.in.in libindicator-12.10.2+14.04.20141007/libindicator/indicator3-0.4.pc.in.in
--- libindicator-12.10.2+14.04.20141007.orig/libindicator/indicator3-0.4.pc.in.in	2014-10-07 07:26:23.000000000 -0700
+++ libindicator-12.10.2+14.04.20141007/libindicator/indicator3-0.4.pc.in.in	2015-12-08 05:07:59.991382263 -0800
@@ -4,7 +4,7 @@ libdir=@libdir@
 bindir=@bindir@
 includedir=@includedir@

-indicatordir=${exec_prefix}/lib/indicators3/@indicator_abi_version@/
+indicatordir=${libdir}/indicators3/@indicator_abi_version@/
 iconsdir=@datarootdir@/@PACKAGE@/icons/

 Cflags: -I${includedir}/libindicator3-0.@indicator_api_version@
