--- /src/ShortcutLabel.vala	2020-04-03 21:43:00.000000000 +0300
+++ /src/ShortcutLabel.vala	2020-04-24 00:34:23.533289691 +0300
@@ -21,6 +21,10 @@
 
     private static Gee.ArrayList<Settings> settings_list;
     private static Settings get_settings_for_schema (string schema_id) {
+        if (settings_list == null) {
+            settings_list = new Gee.ArrayList<Settings> ();
+        }
+
         foreach (var settings in settings_list) {
             if (settings.schema_id == schema_id) {
                 return settings;
@@ -54,10 +58,6 @@
         Object (accels: accels);
     }
 
-    static construct {
-        settings_list = new Gee.ArrayList<Settings> ();
-    }
-
     construct {
         column_spacing = 6;
 
