diff -Naur wingpanel-3.0.3.orig/wingpanel-interface/FocusManager.vala wingpanel-3.0.3/wingpanel-interface/FocusManager.vala
--- wingpanel-3.0.3.orig/wingpanel-interface/FocusManager.vala	2023-05-04 11:42:30.386907019 +0200
+++ wingpanel-3.0.3/wingpanel-interface/FocusManager.vala	2023-05-04 11:43:04.960242822 +0200
@@ -112,7 +112,11 @@
         }
 
         if (window != null) {
+#if HAS_MUTTER44
+            window.begin_grab_op (Meta.GrabOp.MOVING, null, null, time);
+#else
             display.begin_grab_op (window, Meta.GrabOp.MOVING, false, true, button, state, time, x, y);
+#endif
             return true;
         }
 
diff -Naur wingpanel-3.0.3.orig/wingpanel-interface/meson.build wingpanel-3.0.3/wingpanel-interface/meson.build
--- wingpanel-3.0.3.orig/wingpanel-interface/meson.build	2023-05-04 11:42:30.386907019 +0200
+++ wingpanel-3.0.3/wingpanel-interface/meson.build	2023-05-04 11:43:32.430244757 +0200
@@ -5,47 +5,6 @@
 c_flags =['-DCLUTTER_ENABLE_COMPOSITOR_API', '-DCLUTTER_ENABLE_EXPERIMENTAL_API',
 		'-DCOGL_ENABLE_EXPERIMENTAL_API', '-DCOGL_ENABLE_EXPERIMENTAL_2_0_API']
 
-mutter336_dep = dependency('libmutter-6', version: ['>= 3.35.1', '< 3.37'], required: false)
-if mutter336_dep.found()
-	libmutter_dep = dependency('libmutter-6', version: '>= 3.35.1')
-	mutter_dep = [
-		dependency('mutter-cogl-6'), dependency('mutter-cogl-pango-6'),
-		dependency('mutter-cogl-path-6'), dependency('mutter-clutter-6')
-	]
-	vala_flags = []
-endif
-
-mutter338_dep = dependency('libmutter-7', version: ['>= 3.37', '< 3.39'], required: false)
-if mutter338_dep.found()
-	libmutter_dep = dependency('libmutter-7', version: '>= 3.37')
-	mutter_dep = [
-		dependency('mutter-cogl-7'), dependency('mutter-cogl-pango-7'),
-		dependency('mutter-clutter-7')
-	]
-	vala_flags = ['--define', 'HAS_MUTTER338']
-endif
-
-mutter40_dep = dependency('libmutter-8', version: ['>= 40', '< 41'], required: false)
-if mutter40_dep.found()
-	libmutter_dep = dependency('libmutter-8', version: '>= 40')
-	mutter_dep = [
-		dependency('mutter-cogl-8'), dependency('mutter-cogl-pango-8'),
-		dependency('mutter-clutter-8')
-	]
-	vala_flags = ['--define', 'HAS_MUTTER338', '--define', 'HAS_MUTTER40']
-endif
-
-mutter41_dep = dependency('libmutter-9', version: ['>= 41', '< 42'], required: false)
-if mutter41_dep.found()
-	libmutter_dep = dependency('libmutter-9', version: '>= 41')
-	mutter_dep = [
-		libmutter_dep,
-		dependency('mutter-cogl-9'), dependency('mutter-cogl-pango-9'),
-		dependency('mutter-clutter-9')
-	]
-	vala_flags = ['--define', 'HAS_MUTTER338', '--define', 'HAS_MUTTER40', '--define', 'HAS_MUTTER41']
-endif
-
 mutter42_dep = dependency('libmutter-10', version: ['>= 42', '< 43'], required: false)
 if mutter42_dep.found()
 	libmutter_dep = dependency('libmutter-10', version: '>= 42')
@@ -54,7 +13,7 @@
 		dependency('mutter-cogl-10'), dependency('mutter-cogl-pango-10'),
 		dependency('mutter-clutter-10')
 	]
-	vala_flags = ['--define', 'HAS_MUTTER338', '--define', 'HAS_MUTTER40', '--define', 'HAS_MUTTER41', '--define', 'HAS_MUTTER42']
+	vala_flags = []
 endif
 
 mutter43_dep = dependency('libmutter-11', version: ['>= 43', '< 44'], required: false)
@@ -65,7 +24,18 @@
 		dependency('mutter-cogl-11'), dependency('mutter-cogl-pango-11'),
 		dependency('mutter-clutter-11')
 	]
-	vala_flags = ['--define', 'HAS_MUTTER338', '--define', 'HAS_MUTTER40', '--define', 'HAS_MUTTER41', '--define', 'HAS_MUTTER42', '--define', 'HAS_MUTTER43']
+	vala_flags = ['--define', 'HAS_MUTTER43']
+endif
+
+mutter44_dep = dependency('libmutter-12', version: ['>= 44', '< 45'], required: false)
+if mutter44_dep.found()
+	libmutter_dep = dependency('libmutter-12', version: '>= 44')
+	mutter_dep = [
+		libmutter_dep,
+		dependency('mutter-cogl-12'), dependency('mutter-cogl-pango-12'),
+		dependency('mutter-clutter-12')
+	]
+	vala_flags = ['--define', 'HAS_MUTTER43', '--define', 'HAS_MUTTER44']
 endif
 
 if mutter_dep.length() == 0
diff -Naur wingpanel-3.0.3.orig/wingpanel-interface/Utils.vala wingpanel-3.0.3/wingpanel-interface/Utils.vala
--- wingpanel-3.0.3.orig/wingpanel-interface/Utils.vala	2023-05-04 11:42:30.386907019 +0200
+++ wingpanel-3.0.3/wingpanel-interface/Utils.vala	2023-05-04 11:43:17.166910345 +0200
@@ -28,13 +28,8 @@
 
     private class DummyOffscreenEffect : Clutter.OffscreenEffect {
         public signal void done_painting ();
-#if HAS_MUTTER40
         public override void post_paint (Clutter.PaintNode node, Clutter.PaintContext context) {
             base.post_paint (node, context);
-#else
-        public override void post_paint (Clutter.PaintContext context) {
-            base.post_paint (context);
-#endif
             Idle.add (() => {
                 done_painting ();
                 return false;
