=== modified file 'src/Widgets/Panel.vala'
--- src/Widgets/Panel.vala	2012-11-01 22:33:15 +0000
+++ src/Widgets/Panel.vala	2013-02-18 18:56:28 +0000
@@ -80,8 +80,8 @@
         private int panel_x;
         private int panel_y;
         private int panel_width;
-        private uint animation_timer = 0;
-        private int panel_displacement = -40;
+        private uint animation_timer = 1;
+        private int panel_displacement = 0;
 
         private Box container;
         private Box left_wrapper;

