=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2013-12-23 23:32:28 +0000
+++ CMakeLists.txt	2014-01-24 17:20:15 +0000
@@ -71,7 +71,7 @@
 set (LIB_PACKAGES
   glib-2.0
   gio-2.0
-  gee-1.0
+  gee-0.8
   gmodule-2.0
   gtk+-3.0
   gio-unix-2.0
@@ -80,7 +80,7 @@
 set (LIB_PKG
   glib-2.0>=2.32
   gio-2.0
-  gee-1.0
+  gee-0.8
   gmodule-2.0
   gtk+-3.0>=3.10
   gio-unix-2.0

=== modified file 'lib/switchboard-2.0.deps'
--- lib/switchboard-2.0.deps	2013-12-11 23:03:22 +0000
+++ lib/switchboard-2.0.deps	2014-01-24 17:20:15 +0000
@@ -1,6 +1,6 @@
 glib-2.0
 gio-2.0
-gee-1.0
+gee-0.8
 gmodule-2.0
 gtk+-3.0
 gio-unix-2.0

=== modified file 'lib/switchboard.pc.cmake'
--- lib/switchboard.pc.cmake	2013-12-11 23:03:22 +0000
+++ lib/switchboard.pc.cmake	2014-01-24 17:20:15 +0000
@@ -8,5 +8,5 @@
 Version: 2.0  
 Libs: -lswitchboard-2.0
 Cflags: -I@DOLLAR@{includedir}/switchboard-2.0
-Requires: glib-2.0 gio-2.0 gee-1.0 gmodule-2.0 gtk+-3.0 gio-unix-2.0
+Requires: glib-2.0 gio-2.0 gee-0.8 gmodule-2.0 gtk+-3.0 gio-unix-2.0
 

