diff -Naur indicator-sound-12.10.2+14.04.20140401/CMakeLists.txt indicator-sound-elementary/CMakeLists.txt
--- indicator-sound-12.10.2+14.04.20140401/CMakeLists.txt	2014-04-02 01:19:46.000000000 +0200
+++ indicator-sound-elementary/CMakeLists.txt	2016-02-04 16:12:14.374512000 +0100
@@ -81,5 +81,3 @@
 set (GTEST_INCLUDE_DIR ${GTEST_SOURCE_DIR}/..)
 set (GTEST_LIBS -lpthread)
 
-add_subdirectory(tests)
-
diff -Naur indicator-sound-12.10.2+14.04.20140401/data/indicator-sound.desktop.in indicator-sound-elementary/data/indicator-sound.desktop.in
--- indicator-sound-12.10.2+14.04.20140401/data/indicator-sound.desktop.in	2014-04-02 01:19:46.000000000 +0200
+++ indicator-sound-elementary/data/indicator-sound.desktop.in	2016-02-04 16:12:14.374512000 +0100
@@ -2,7 +2,7 @@
 Type=Application
 Name=Indicator Sound
 Exec=@CMAKE_INSTALL_FULL_LIBEXECDIR@/indicator-sound/indicator-sound-service
-OnlyShowIn=Unity;XFCE;GNOME;
+OnlyShowIn=Unity;XFCE;GNOME;Pantheon;
 NoDisplay=true
 StartupNotify=false
 Terminal=false
diff -Naur indicator-sound-12.10.2+14.04.20140401/src/service.vala indicator-sound-elementary/src/service.vala
--- indicator-sound-12.10.2+14.04.20140401/src/service.vala	2014-04-02 01:20:41.000000000 +0200
+++ indicator-sound-elementary/src/service.vala	2016-02-04 16:49:32.169640511 +0100
@@ -238,15 +238,15 @@
 		double volume = this.volume_control.get_volume ();
 		string icon;
 		if (this.volume_control.mute)
-			icon = this.mute_blocks_sound ? "audio-volume-muted-blocking-panel" : "audio-volume-muted-panel";
+			icon = "audio-volume-muted-symbolic";
 		else if (volume <= 0.0)
-			icon = "audio-volume-low-zero-panel";
+			icon = "audio-volume-muted-symbolic";
 		else if (volume <= 0.3)
-			icon = "audio-volume-low-panel";
+			icon = "audio-volume-low-symbolic";
 		else if (volume <= 0.7)
-			icon = "audio-volume-medium-panel";
+			icon = "audio-volume-medium-symbolic";
 		else
-			icon  = "audio-volume-high-panel";
+			icon = "audio-volume-high-symbolic";
 
 		string accessible_name;
 		if (this.volume_control.mute) {
