diff -Naur indicator-sound-12.10.2+15.10.20150915/CMakeLists.txt indicator-sound-patched/CMakeLists.txt
--- indicator-sound-12.10.2+15.10.20150915/CMakeLists.txt	2015-09-15 10:19:41.000000000 +0200
+++ indicator-sound-patched/CMakeLists.txt	2016-02-19 21:50:19.738548797 +0100
@@ -81,5 +81,3 @@
 set (GTEST_INCLUDE_DIR ${GTEST_SOURCE_DIR}/..)
 set (GTEST_LIBS -lpthread)
 
-add_subdirectory(tests)
-
diff -Naur indicator-sound-12.10.2+15.10.20150915/src/service.vala indicator-sound-patched/src/service.vala
--- indicator-sound-12.10.2+15.10.20150915/src/service.vala	2015-09-15 10:19:47.000000000 +0200
+++ indicator-sound-patched/src/service.vala	2016-02-19 21:53:56.690153280 +0100
@@ -247,15 +247,15 @@
 		double volume = this.volume_control.volume.volume;
 		string icon;
 		if (this.volume_control.mute || volume <= 0.0)
-			icon = this.mute_blocks_sound ? "audio-volume-muted-blocking-panel" : "audio-volume-muted-panel";
+			icon = this.mute_blocks_sound ? "audio-volume-muted-symbolic" : "audio-volume-muted-symbolic";
 		else if (this.accounts_service != null && this.accounts_service.silentMode)
-		  icon = "audio-volume-muted-panel";
+		  icon = "audio-volume-muted-symbolic";
 		else if (volume <= 0.3)
-			icon = "audio-volume-low-panel";
+			icon = "audio-volume-low-symbolic";
 		else if (volume <= 0.7)
-			icon = "audio-volume-medium-panel";
+			icon = "audio-volume-medium-symbolic";
 		else
-			icon  = "audio-volume-high-panel";
+			icon  = "audio-volume-high-symbolic";
 
 		string accessible_name;
 		if (this.volume_control.mute) {
