diff -Naur indicator-sound-12.10.2+14.04.20131125.orig/data/CMakeLists.txt indicator-sound-12.10.2+14.04.20131125/data/CMakeLists.txt
--- indicator-sound-12.10.2+14.04.20131125.orig/data/CMakeLists.txt	2014-01-09 15:30:53.694202029 +0100
+++ indicator-sound-12.10.2+14.04.20131125/data/CMakeLists.txt	2014-01-09 15:31:19.199320537 +0100
@@ -17,26 +17,6 @@
 )
 
 ###########################
-# Upstart Job
-###########################
-
-set(
-  INDICATOR_SOUND_CONF
-  "${CMAKE_CURRENT_BINARY_DIR}/indicator-sound.conf"
-)
-
-configure_file(
-  "indicator-sound.conf.in"
-  ${INDICATOR_SOUND_CONF}
-  @ONLY
-)
-
-install(
-  FILES "${INDICATOR_SOUND_CONF}"
-  DESTINATION "${CMAKE_INSTALL_DATADIR}/upstart/sessions/"
-)
-
-###########################
 # XDG Autostart
 ###########################
 
