diff -Naur indicator-sound-12.10.2+14.04.20131125.orig/CMakeLists.txt indicator-sound-12.10.2+14.04.20131125/CMakeLists.txt
--- indicator-sound-12.10.2+14.04.20131125.orig/CMakeLists.txt	2013-12-03 17:27:11.255597929 +0100
+++ indicator-sound-12.10.2+14.04.20131125/CMakeLists.txt	2013-12-03 17:27:27.210689345 +0100
@@ -25,13 +25,11 @@
 
 set(PULSE_AUDIO_REQUIRED_VERSION 0.9.19)
 set(GIO_2_0_REQUIRED_VERSION 2.25.13)
-set(URL_DISPATCHER_1_REQUIRED_VERSION 1)
 
 pkg_check_modules(
   PULSEAUDIO REQUIRED
   libpulse-mainloop-glib>=${PULSE_AUDIO_REQUIRED_VERSION}
   gio-unix-2.0>=${GIO_2_0_REQUIRED_VERSION}
-  url-dispatcher-1>=${URL_DISPATCHER_1_REQUIRED_VERSION}
 )
 include_directories(${PULSEAUDIO_INCLUDE_DIRS})
 
diff -Naur indicator-sound-12.10.2+14.04.20131125.orig/src/CMakeLists.txt indicator-sound-12.10.2+14.04.20131125/src/CMakeLists.txt
--- indicator-sound-12.10.2+14.04.20131125.orig/src/CMakeLists.txt	2013-12-03 17:27:11.256597935 +0100
+++ indicator-sound-12.10.2+14.04.20131125/src/CMakeLists.txt	2013-12-03 17:27:51.698829658 +0100
@@ -21,7 +21,6 @@
     --vapidir=${CMAKE_SOURCE_DIR}/vapi/
     --vapidir=.
     --target-glib=2.36
-    --pkg=url-dispatcher
     --pkg=bus-watcher
 )
 
diff -Naur indicator-sound-12.10.2+14.04.20131125.orig/src/service.vala indicator-sound-12.10.2+14.04.20131125/src/service.vala
--- indicator-sound-12.10.2+14.04.20131125.orig/src/service.vala	2013-12-03 17:27:11.256597935 +0100
+++ indicator-sound-12.10.2+14.04.20131125/src/service.vala	2013-12-03 17:28:39.474103428 +0100
@@ -36,7 +36,6 @@
 		this.menus = new HashTable<string, SoundMenu> (str_hash, str_equal);
 		this.menus.insert ("desktop_greeter", new SoundMenu (null, SoundMenu.DisplayFlags.SHOW_MUTE));
 		this.menus.insert ("desktop", new SoundMenu ("indicator.desktop-settings", SoundMenu.DisplayFlags.SHOW_MUTE));
-		this.menus.insert ("phone", new SoundMenu ("indicator.phone-settings", SoundMenu.DisplayFlags.HIDE_INACTIVE_PLAYERS));
 
 		this.menus.@foreach ( (profile, menu) => {
 			this.volume_control.bind_property ("active-mic", menu, "show-mic-volume", BindingFlags.SYNC_CREATE);
@@ -75,7 +74,6 @@
 		{ "root", null, null, "@a{sv} {}", null },
 		{ "scroll", activate_scroll_action, "i", null, null },
 		{ "desktop-settings", activate_desktop_settings, null, null, null },
-		{ "phone-settings", activate_phone_settings, null, null, null },
 	};
 
 	MainLoop loop;
@@ -132,10 +130,6 @@
 		}
 	}
 
-	void activate_phone_settings (SimpleAction action, Variant? param) {
-		UrlDispatch.send ("settings:///system/sound");
-	}
-
 	/* Returns a serialized version of @icon_name suited for the panel */
 	static Variant serialize_themed_icon (string icon_name)
 	{
diff -Naur indicator-sound-12.10.2+14.04.20131125.orig/vapi/url-dispatcher.vapi indicator-sound-12.10.2+14.04.20131125/vapi/url-dispatcher.vapi
--- indicator-sound-12.10.2+14.04.20131125.orig/vapi/url-dispatcher.vapi	2013-12-03 17:27:11.256597935 +0100
+++ indicator-sound-12.10.2+14.04.20131125/vapi/url-dispatcher.vapi	1970-01-01 01:00:00.000000000 +0100
@@ -1,9 +0,0 @@
-[CCode (cprefix="", lower_case_cprefix="", cheader_filename="liburl-dispatcher-1/url-dispatcher.h")]
-
-namespace UrlDispatch
-{
-  public delegate void DispatchCallback ();
-
-  [CCode (cname = "url_dispatch_send")]
-  public static void send (string url, DispatchCallback? func = null);
-}
