diff -Naur terminal-6.1.2.orig/src/meson.build terminal-6.1.2/src/meson.build
--- terminal-6.1.2.orig/src/meson.build	2023-04-08 16:50:59.116347583 +0200
+++ terminal-6.1.2/src/meson.build	2023-04-08 16:52:48.453019270 +0200
@@ -38,33 +38,3 @@
     install : true
 )
 
-# tests
-
-test_env = [
-    'G_TEST_SRCDIR=' + meson.current_source_dir(),
-    'G_TEST_BUILDDIR=' + meson.current_build_dir(),
-    'GSETTINGS_SCHEMA_DIR=' + meson.project_build_root() / 'data',
-    'GSETTINGS_BACKEND=memory',
-    'GIO_USE_VFS=local'
-]
-
-add_test_setup(
-  'headless',
-  exe_wrapper: [ find_program('xvfb-run'), '-a', '-s', '-screen 0 1024x768x24 -noreset' ],
-  is_default: true
-)
-
-app_test = executable(
-    meson.project_name() + '.tests.application',
-    terminal_sources + 'tests/Application.vala',
-    dependencies: terminal_deps,
-    vala_args: [ '--define', 'TESTS']
-)
-
-test(
-    'Application',
-    app_test,
-    env: test_env,
-    protocol: 'tap',
-    depends: test_schemas
-)
