diff --git a/plugins/brackets-completion/brackets-completion.vala b/plugins/brackets-completion/brackets-completion.vala
index 5639b4460e..ba3727e80c 100644
--- a/plugins/brackets-completion/brackets-completion.vala
+++ b/plugins/brackets-completion/brackets-completion.vala
@@ -9,7 +9,7 @@ public class Scratch.Plugins.BracketsCompletion : Peas.ExtensionBase, Scratch.Se
         "", " ", "\b", "\r", "\n", "\t", ",", ".", ";", ":"
     };
 
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
 
     private Gee.HashMap<string, string> brackets;
     private Gee.HashMap<uint, string> keys;
diff --git a/plugins/detect-indent/detect-indent.vala b/plugins/detect-indent/detect-indent.vala
index 1e267685e5..2c7dfb84b9 100644
--- a/plugins/detect-indent/detect-indent.vala
+++ b/plugins/detect-indent/detect-indent.vala
@@ -2,7 +2,7 @@ public class Scratch.Plugins.DetectIndent: Peas.ExtensionBase, Scratch.Services.
     const int MAX_LINES = 500;
 
     Scratch.Services.Interface plugins;
-    public Object object {owned get; construct;}
+    public Object object {owned get; set construct;}
 
     public void update_state () {
     }
diff --git a/plugins/editorconfig/editorconfig.vala b/plugins/editorconfig/editorconfig.vala
index 8286cf0b6a..d7cabe1c78 100644
--- a/plugins/editorconfig/editorconfig.vala
+++ b/plugins/editorconfig/editorconfig.vala
@@ -19,7 +19,7 @@
 
 public class Scratch.Plugins.EditorConfigPlugin: Peas.ExtensionBase, Scratch.Services.ActivatablePlugin {
     Scratch.Services.Interface plugins;
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
     private Code.FormatBar format_bar;
 
     public void update_state () { }
diff --git a/plugins/fuzzy-search/fuzzy-search.vala b/plugins/fuzzy-search/fuzzy-search.vala
index c28a54450e..41ebeec144 100644
--- a/plugins/fuzzy-search/fuzzy-search.vala
+++ b/plugins/fuzzy-search/fuzzy-search.vala
@@ -7,7 +7,7 @@
 
 
 public class Scratch.Plugins.FuzzySearch: Peas.ExtensionBase, Scratch.Services.ActivatablePlugin {
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
     private const uint ACCEL_KEY = Gdk.Key.F;
     private const Gdk.ModifierType ACCEL_MODTYPE = Gdk.ModifierType.MOD1_MASK;
 
diff --git a/plugins/highlight-word-selection/highlight-word-selection.vala b/plugins/highlight-word-selection/highlight-word-selection.vala
index 91303df71d..15e13c9a4e 100644
--- a/plugins/highlight-word-selection/highlight-word-selection.vala
+++ b/plugins/highlight-word-selection/highlight-word-selection.vala
@@ -26,7 +26,7 @@ public class Scratch.Plugins.HighlightSelectedWords : Peas.ExtensionBase, Scratc
     private const uint SELECTION_HIGHLIGHT_MAX_CHARS = 255;
 
     Scratch.Services.Interface plugins;
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
 
     public void update_state () {}
 
diff --git a/plugins/markdown-actions/markdown-actions.vala b/plugins/markdown-actions/markdown-actions.vala
index 753e2031f8..a446edaee7 100644
--- a/plugins/markdown-actions/markdown-actions.vala
+++ b/plugins/markdown-actions/markdown-actions.vala
@@ -22,7 +22,7 @@ public class Code.Plugins.MarkdownActions : Peas.ExtensionBase, Scratch.Services
     Scratch.Widgets.SourceView current_source;
     Scratch.Services.Interface plugins;
 
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
 
     public void update_state () {}
 
diff --git a/plugins/pastebin/pastebin.vala b/plugins/pastebin/pastebin.vala
index 2d32634904..7923d6eba9 100644
--- a/plugins/pastebin/pastebin.vala
+++ b/plugins/pastebin/pastebin.vala
@@ -78,7 +78,7 @@ namespace Scratch.Services {
 public class Scratch.Plugins.Pastebin : Peas.ExtensionBase, Scratch.Services.ActivatablePlugin {
     GLib.MenuItem? menuitem = null;
     GLib.Menu? share_menu = null;
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
 
     Scratch.Services.Document? doc = null;
     Scratch.Services.Interface plugins;
diff --git a/plugins/preserve-indent/preserve-indent.vala b/plugins/preserve-indent/preserve-indent.vala
index bfb8ff801b..37a0debfac 100644
--- a/plugins/preserve-indent/preserve-indent.vala
+++ b/plugins/preserve-indent/preserve-indent.vala
@@ -26,7 +26,7 @@ public class Scratch.Plugins.PreserveIndent : Peas.ExtensionBase, Scratch.Servic
     private int last_clipboard_indent_level = 0;
     private bool waiting_for_clipboard_text = false;
 
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
 
     public void activate () {
         this.documents = new Gee.TreeSet<weak Services.Document> ();
diff --git a/plugins/spell/spell.vala b/plugins/spell/spell.vala
index f27bae823f..5f9ee75435 100644
--- a/plugins/spell/spell.vala
+++ b/plugins/spell/spell.vala
@@ -29,7 +29,7 @@ public class Scratch.Plugins.Spell: Peas.ExtensionBase, Scratch.Services.Activat
     GtkSpell.Checker spell = null;
 #endif
 
-    public Object object {owned get; construct;}
+    public Object object {owned get; set construct;}
 
     public void update_state () {
     }
diff --git a/plugins/vim-emulation/vim-emulation.vala b/plugins/vim-emulation/vim-emulation.vala
index 426737da21..394c5e5c8a 100644
--- a/plugins/vim-emulation/vim-emulation.vala
+++ b/plugins/vim-emulation/vim-emulation.vala
@@ -34,7 +34,7 @@ public class Scratch.Plugins.VimEmulation : Peas.ExtensionBase, Scratch.Services
     Scratch.Widgets.SourceView? view = null;
 
     Scratch.Services.Interface plugins;
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
 
     construct {
         views = new Gee.TreeSet<Scratch.Widgets.SourceView> ();
diff --git a/plugins/word-completion/plugin.vala b/plugins/word-completion/plugin.vala
index 176dd5ece1..7719a5c2f4 100644
--- a/plugins/word-completion/plugin.vala
+++ b/plugins/word-completion/plugin.vala
@@ -21,7 +21,7 @@
 public class Scratch.Plugins.Completion : Peas.ExtensionBase, Scratch.Services.ActivatablePlugin {
     public const uint INTERACTIVE_DELAY = 500;
 
-    public Object object { owned get; construct; }
+    public Object object { owned get; set construct; }
 
     private List<Gtk.SourceView> text_view_list = new List<Gtk.SourceView> ();
     public Euclide.Completion.Parser parser {get; private set;}
diff --git a/src/Services/PluginManager.vala b/src/Services/PluginManager.vala
index 74c5d1e9cd..9b4ee1e229 100644
--- a/src/Services/PluginManager.vala
+++ b/src/Services/PluginManager.vala
@@ -9,7 +9,7 @@ public interface Scratch.Services.ActivatablePlugin : Object {
     public abstract void activate ();
     public abstract void deactivate ();
     public virtual void update_state () {}
-    public abstract GLib.Object object { owned get; construct; }
+    public abstract GLib.Object object { owned get; set construct; }
 }
 
 // Object shared with plugins providing signals and methods to interface with application
