=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2013-01-18 13:07:36 +0000
+++ CMakeLists.txt	2013-04-04 09:28:30 +0000
@@ -57,24 +57,6 @@
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/maya-calendar.desktop DESTINATION "${CMAKE_INSTALL_PREFIX}/share/applications")
 
 #
-# i18n
-#
-find_package (Gettext REQUIRED)
-
-add_custom_command (OUTPUT ${CMAKE_SOURCE_DIR}/po/maya.pot
-        COMMAND xgettext -c --files-from ${CMAKE_SOURCE_DIR}/po/POTFILES.in --keyword=_ -o ${CMAKE_SOURCE_DIR}/po/maya.pot --no-wrap --no-location
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-        )
-
-if (GETTEXT_FOUND)
-	message(STATUS GETTEXT_FOUND = ${GETTEXT_FOUND})
-        set (HAVE_GETTEXT true)
-        file (GLOB _translations ${CMAKE_SOURCE_DIR}/po/*.po)
-        GETTEXT_CREATE_TRANSLATIONS (${CMAKE_SOURCE_DIR}/po/${PROJECT_NAME}.pot
-                ALL ${_translations})
-endif (GETTEXT_FOUND)
-
-#
 # Maya
 #
 find_package(PkgConfig)

