=== modified file 'core/Model/CalendarModel.vala'
--- core/Model/CalendarModel.vala	2015-01-16 23:38:19 +0000
+++ core/Model/CalendarModel.vala	2015-09-28 18:16:17 +0000
@@ -362,7 +362,7 @@
         try {
             var cancellable = new GLib.Cancellable ();
             connecting (source, cancellable);
-            var client = new E.CalClient.connect_sync (source, E.CalClientSourceType.EVENTS, cancellable);
+            var client = new E.CalClient.connect_sync (source, E.CalClientSourceType.EVENTS, 30, cancellable);
             source_client.insert (source.dup_uid (), client);
         } catch (Error e) {
             error_received (e.message);

=== modified file 'vapi/libecalendar-1.2.vapi'
--- vapi/libecalendar-1.2.vapi	2014-12-31 16:11:57 +0000
+++ vapi/libecalendar-1.2.vapi	2015-09-28 18:17:04 +0000
@@ -24,7 +24,7 @@
 		public CalClient (E.Source source, E.CalClientSourceType source_type) throws GLib.Error;
 		public virtual async void connect (E.Source source, E.CalClientSourceType source_type, GLib.Cancellable? cancellable) throws GLib.Error;
 		[CCode (has_construct_function = false, cname = "e_cal_client_connect_sync")]
-		public CalClient.connect_sync (E.Source source, E.CalClientSourceType source_type, GLib.Cancellable? cancellable = null) throws GLib.Error;
+		public CalClient.connect_sync (E.Source source, E.CalClientSourceType source_type, int32 wait_for_connected_seconds, GLib.Cancellable? cancellable = null) throws GLib.Error;
 		public async bool add_timezone (iCal.TimeZone zone, GLib.Cancellable cancellable) throws GLib.Error;
 		public bool add_timezone_sync (iCal.TimeZone zone, GLib.Cancellable cancellable) throws GLib.Error;
 		public bool check_one_alarm_only ();

