From cd867b1158f5fca433e7920aa84eb4e6f7080d7b Mon Sep 17 00:00:00 2001
From: William Jon McCann <jmccann@redhat.com>
Date: Tue, 08 May 2012 20:14:57 +0000
Subject: Adapt to gnome-desktop API change for display labels

GdkColor -> GdkRGBA
---
Index: gnome-control-center-3.4.2/configure.ac
===================================================================
--- gnome-control-center-3.4.2.orig/configure.ac	2012-06-11 11:32:30.000000000 +0200
+++ gnome-control-center-3.4.2/configure.ac	2012-06-11 11:34:33.246200700 +0200
@@ -84,7 +84,7 @@
 GSD_REQUIRED_VERSION=3.3.91
 NETWORK_MANAGER_REQUIRED_VERSION=0.8.992
 LIBNOTIFY_REQUIRED_VERSION=0.7.3
-GNOME_DESKTOP_REQUIRED_VERSION=3.1.91
+GNOME_DESKTOP_REQUIRED_VERSION=3.5.1
 SCHEMAS_REQUIRED_VERSION=3.3.0
 LIBWACOM_REQUIRED_VERSION=0.3
 
Index: gnome-control-center-3.4.2/panels/display/cc-display-panel.c
===================================================================
--- gnome-control-center-3.4.2.orig/panels/display/cc-display-panel.c	2012-06-11 11:32:30.000000000 +0200
+++ gnome-control-center-3.4.2/panels/display/cc-display-panel.c	2012-06-11 11:34:33.250200700 +0200
@@ -657,7 +657,7 @@
 rebuild_current_monitor_label (CcDisplayPanel *self)
 {
   char *str, *tmp;
-  GdkColor color;
+  GdkRGBA color;
   gboolean use_color;
 
   if (self->priv->current_output)
@@ -668,7 +668,7 @@
         tmp = g_strdup (gnome_rr_output_info_get_display_name (self->priv->current_output));
 
       str = g_strdup_printf ("<b>%s</b>", tmp);
-      gnome_rr_labeler_get_color_for_output (self->priv->labeler, self->priv->current_output, &color);
+      gnome_rr_labeler_get_rgba_for_output (self->priv->labeler, self->priv->current_output, &color);
       use_color = TRUE;
       g_free (tmp);
     }
@@ -684,15 +684,10 @@
   if (use_color)
     {
       GdkRGBA black = { 0, 0, 0, 1.0 };
-      GdkRGBA light;
 
-      light.red = color.red / 65535.0;
-      light.green = color.green / 65535.0;
-      light.blue = color.blue / 65535.0;
-      light.alpha = 1.0;
       gtk_widget_override_background_color (self->priv->current_monitor_event_box,
                                             gtk_widget_get_state_flags (self->priv->current_monitor_event_box),
-                                            &light);
+                                            &color);
 
       /* Make the label explicitly black.  We don't want it to follow the
        * theme's colors, since the label is always shown against a light
@@ -2003,7 +1998,7 @@
   PangoLayout *layout = get_display_name (self, output);
   PangoRectangle ink_extent, log_extent;
   GdkRectangle viewport;
-  GdkColor output_color;
+  GdkRGBA output_color;
   double r, g, b;
   double available_w;
   double factor;
@@ -2068,10 +2063,10 @@
   cairo_rectangle (cr, x, y, w * scale + 0.5, h * scale + 0.5);
   cairo_clip_preserve (cr);
 
-  gnome_rr_labeler_get_color_for_output (self->priv->labeler, output, &output_color);
-  r = output_color.red / 65535.0;
-  g = output_color.green / 65535.0;
-  b = output_color.blue / 65535.0;
+  gnome_rr_labeler_get_rgba_for_output (self->priv->labeler, output, &output_color);
+  r = output_color.red;
+  g = output_color.green;
+  b = output_color.blue;
 
   if (!gnome_rr_output_info_is_active (output))
     {
