diff -Naur gnome-control-center-3.4.2/panels/bluetooth/cc-bluetooth-panel.c gnome-control-center-3.4.2.patched/panels/bluetooth/cc-bluetooth-panel.c
--- gnome-control-center-3.4.2/panels/bluetooth/cc-bluetooth-panel.c	2012-03-05 15:04:55.000000000 +0100
+++ gnome-control-center-3.4.2.patched/panels/bluetooth/cc-bluetooth-panel.c	2013-04-04 19:33:21.829451783 +0200
@@ -347,7 +347,7 @@
 	state = gtk_switch_get_active (GTK_SWITCH (WID ("switch_bluetooth")));
 	g_debug ("Power switched to %s", state ? "off" : "on");
 	bluetooth_killswitch_set_state (self->priv->killswitch,
-					state ? KILLSWITCH_STATE_UNBLOCKED : KILLSWITCH_STATE_SOFT_BLOCKED);
+					state ? BLUETOOTH_KILLSWITCH_STATE_UNBLOCKED : BLUETOOTH_KILLSWITCH_STATE_SOFT_BLOCKED);
 }
 
 static void
@@ -369,7 +369,7 @@
 static void
 cc_bluetooth_panel_update_power (CcBluetoothPanel *self)
 {
-	KillswitchState state;
+	BluetoothKillswitchState state;
 	char *path;
 	gboolean powered, sensitive;
 	GtkSwitch *button;
@@ -387,12 +387,12 @@
 
 	if (path == NULL &&
 	    bluetooth_killswitch_has_killswitches (self->priv->killswitch) &&
-	    state != KILLSWITCH_STATE_HARD_BLOCKED) {
+	    state != BLUETOOTH_KILLSWITCH_STATE_HARD_BLOCKED) {
 		g_debug ("Default adapter is unpowered, but should be available");
 		sensitive = TRUE;
 		cc_bluetooth_panel_update_treeview_message (self, _("Bluetooth is disabled"));
 	} else if (path == NULL &&
-		   state == KILLSWITCH_STATE_HARD_BLOCKED) {
+		   state == BLUETOOTH_KILLSWITCH_STATE_HARD_BLOCKED) {
 		g_debug ("Bluetooth is Hard blocked");
 		sensitive = FALSE;
 		cc_bluetooth_panel_update_treeview_message (self, _("Bluetooth is disabled by hardware switch"));
@@ -705,7 +705,7 @@
 
 static void
 killswitch_changed (BluetoothKillswitch *killswitch,
-		    KillswitchState      state,
+		    BluetoothKillswitchState      state,
 		    CcBluetoothPanel    *self)
 {
 	cc_bluetooth_panel_update_state (self);
