--- panels/datetime/Makefile.am~	2013-03-15 12:59:32.000000000 +0100
+++ panels/datetime/Makefile.am	2013-03-27 13:00:45.269448422 +0100
@@ -24,6 +24,7 @@
 	$(DATETIME_PANEL_CFLAGS)			\
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""	\
 	-DGNOMECC_DATA_DIR="\"$(pkgdatadir)\""		\
+	-DREALDATADIR="\"$(datadir)\""                  \
 	$(NULL)
 
 noinst_PROGRAMS = test-timezone-gfx test-endianess test-timezone
--- panels/printers/Makefile.am~	2013-03-15 12:59:32.000000000 +0100
+++ panels/printers/Makefile.am	2013-03-27 13:01:55.161013404 +0100
@@ -7,6 +7,9 @@
 	$(SMBCLIENT_CFLAGS)				\
 	-I$(top_srcdir)/shell/				\
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""	\
+	-DREALDATADIR="\"$(datadir)\""                  \
+	-DPREFIX="\"$(prefix)\""                        \
+	-DLIBEXECDIR="\"$(libexecdir)\""                \
 	$(NULL)
 
 noinst_LTLIBRARIES = libprinters.la
