diff -Naur indicator-datetime-12.10.3daily13.06.19.orig/src/indicator-datetime.c indicator-datetime-12.10.3daily13.06.19/src/indicator-datetime.c
--- indicator-datetime-12.10.3daily13.06.19.orig/src/indicator-datetime.c	2013-11-28 16:20:50.177234092 +0100
+++ indicator-datetime-12.10.3daily13.06.19/src/indicator-datetime.c	2013-11-28 16:33:47.650472022 +0100
@@ -881,8 +881,9 @@
 /* Does a quick meausre of how big the string is in
    pixels with a Pango layout */
 static gint
-measure_string (GtkStyle * style, PangoContext * context, const gchar * string)
+measure_string (GtkStyleContext * style, PangoContext * context, const gchar * string)
 {
+	const PangoFontDescription *font_desc;
 	PangoLayout * layout = pango_layout_new(context);
 
 	if (pango_parse_markup(string, -1, 0, NULL, NULL, NULL, NULL))
@@ -890,7 +891,8 @@
 	else
 		pango_layout_set_text(layout, string, -1);
 
-	pango_layout_set_font_description(layout, style->font_desc);
+	gtk_style_context_get(style, GTK_STATE_FLAG_NORMAL, GTK_STYLE_PROPERTY_FONT, &font_desc, NULL);
+	pango_layout_set_font_description(layout, font_desc);
 
 	gint width;
 	pango_layout_get_pixel_size(layout, &width, NULL);
@@ -1056,7 +1058,7 @@
 	/* This is during startup. */
 	if (self->priv->label == NULL) return;
 
-	GtkStyle * style = gtk_widget_get_style(GTK_WIDGET(self->priv->label));
+	GtkStyleContext * style = gtk_widget_get_style_context(GTK_WIDGET(self->priv->label));
 	PangoContext * context = gtk_widget_get_pango_context(GTK_WIDGET(self->priv->label));
 	gint * max_width = &(self->priv->max_width);
 	gint posibilitymask = generate_strftime_bitmask(self->priv->time_string);
@@ -1279,7 +1281,7 @@
 	mi_data->label = gtk_label_new(dbusmenu_menuitem_property_get(newitem, APPOINTMENT_MENUITEM_PROP_LABEL));
 	gtk_misc_set_alignment(GTK_MISC(mi_data->label), 0.0, 0.5);
 	
-	GtkStyle * style = gtk_widget_get_style(GTK_WIDGET(mi_data->label));
+	GtkStyleContext * style = gtk_widget_get_style_context(GTK_WIDGET(mi_data->label));
 	PangoContext * context = gtk_widget_get_pango_context(GTK_WIDGET(mi_data->label));
 	gint length = measure_string(style, context, "MMMMMMMMMMMMMMM"); // 15 char wide string max
 	gtk_widget_set_size_request(GTK_WIDGET(mi_data->label), length, -1); // Set the min size in pixels
