=== modified file 'src/datetime-prefs-locations.c'
--- src/datetime-prefs-locations.c	2013-09-07 12:10:53 +0000
+++ src/datetime-prefs-locations.c	2013-10-26 20:37:14 +0000
@@ -96,7 +96,9 @@
                           COL_ZONE, &zone,
                           COL_VISIBLE_NAME, &name,
                           -1);
-      list = g_slist_prepend (list, time_location_new (zone, name, pos++, now));
+
+      if (zone && name)
+        list = g_slist_prepend (list, time_location_new (zone, name, pos++, now));
 
       g_free (name);
       g_free (zone);

=== modified file 'src/main.c'
--- src/main.c	2013-10-17 22:48:06 +0000
+++ src/main.c	2013-10-28 16:19:49 +0000
@@ -50,6 +50,12 @@
   IndicatorDatetimeService * service;
   GMainLoop * loop;
 
+  /* Work around a deadlock in glib's type initialization. It can be
+   * removed when https://bugzilla.gnome.org/show_bug.cgi?id=674885 is
+   * fixed.
+   */
+  g_type_ensure (G_TYPE_DBUS_CONNECTION);
+
   /* boilerplate i18n */
   setlocale (LC_ALL, "");
   bindtextdomain (GETTEXT_PACKAGE, GNOMELOCALEDIR);

