=== modified file 'src/timezone.c'
--- src/timezone.c	2013-09-09 17:56:22 +0000
+++ src/timezone.c	2013-10-31 00:28:26 +0000
@@ -124,8 +124,11 @@
 
   if (g_strcmp0 (p->timezone->str, timezone))
     {
-      g_string_assign (p->timezone, timezone);
-      g_debug ("%s new timezone set: '%s'", G_STRLOC, timezone);
+      if (timezone != NULL)
+        g_string_assign (p->timezone, timezone);
+      else
+        g_string_set_size (p->timezone, 0);
+      g_debug ("%s new timezone set: '%s'", G_STRLOC, p->timezone->str);
       g_object_notify_by_pspec (G_OBJECT(self), properties[PROP_TIMEZONE]);
     }
 }

