diff -Naur indicator-datetime-13.10.0+14.04.20140415.3/CMakeLists.txt indicator-datetime-13.10.0+14.04.20140415.3.patched/CMakeLists.txt
--- indicator-datetime-13.10.0+14.04.20140415.3/CMakeLists.txt	2014-04-16 00:25:17.000000000 +0200
+++ indicator-datetime-13.10.0+14.04.20140415.3.patched/CMakeLists.txt	2014-12-31 16:06:27.955719560 +0100
@@ -39,7 +39,6 @@
                    libedataserver-1.2>=3.5
                    libcanberra>=0.12
                    libnotify>=0.7.6
-                   url-dispatcher-1>=1
                    properties-cpp>=0.0.1)
 include_directories (SYSTEM ${SERVICE_DEPS_INCLUDE_DIRS})
 
diff -Naur indicator-datetime-13.10.0+14.04.20140415.3/src/actions-live.cpp indicator-datetime-13.10.0+14.04.20140415.3.patched/src/actions-live.cpp
--- indicator-datetime-13.10.0+14.04.20140415.3/src/actions-live.cpp	2014-04-16 00:25:17.000000000 +0200
+++ indicator-datetime-13.10.0+14.04.20140415.3.patched/src/actions-live.cpp	2014-12-31 16:06:27.955719560 +0100
@@ -18,9 +18,6 @@
  */
 
 #include <datetime/actions-live.h>
-
-#include <url-dispatcher.h>
-
 #include <glib.h>
 
 namespace unity {
@@ -52,7 +49,6 @@
 void LiveActions::dispatch_url(const std::string& url)
 {
     g_debug("Dispatching url '%s'", url.c_str());
-    url_dispatch_send(url.c_str(), nullptr, nullptr);
 }
 
 /***
@@ -131,13 +127,11 @@
 
 void LiveActions::phone_open_alarm_app()
 {
-    dispatch_url("appid://com.ubuntu.clock/clock/current-user-version");
 }
 
 void LiveActions::phone_open_appointment(const Appointment& appt)
 {
-    if (!appt.url.empty())
-        dispatch_url(appt.url);
+    if (!appt.url.empty()) {}
     else if (appt.has_alarms)
         phone_open_alarm_app();
     else
@@ -147,12 +141,10 @@
 void LiveActions::phone_open_calendar_app(const DateTime&)
 {
     // does calendar app have a mechanism for specifying dates?
-    dispatch_url("appid://com.ubuntu.calendar/calendar/current-user-version");
 }
 
 void LiveActions::phone_open_settings_app()
 {
-    dispatch_url("settings:///system/time-date");
 }
 
 /***
diff -Naur indicator-datetime-13.10.0+14.04.20140415.3/src/main.cpp indicator-datetime-13.10.0+14.04.20140415.3.patched/src/main.cpp
--- indicator-datetime-13.10.0+14.04.20140415.3/src/main.cpp	2014-04-16 00:25:17.000000000 +0200
+++ indicator-datetime-13.10.0+14.04.20140415.3.patched/src/main.cpp	2014-12-31 16:06:27.955719560 +0100
@@ -35,8 +35,6 @@
 #include <glib/gi18n.h> // bindtextdomain()
 #include <gio/gio.h>
 
-#include <url-dispatcher.h>
-
 #include <locale.h>
 #include <cstdlib> // exit()
 
@@ -83,16 +81,8 @@
     ClockWatcherImpl clock_watcher(live_clock, upcoming_planner);
     Snap snap;
     clock_watcher.alarm_reached().connect([&snap](const Appointment& appt){
-        auto snap_show = [](const Appointment& a){
-            const char* url;
-            if(!a.url.empty())
-                url = a.url.c_str();
-            else // alarm doesn't have a URl associated with it; use a fallback
-                url = "appid://com.ubuntu.clock/clock/current-user-version";
-            url_dispatch_send(url, nullptr, nullptr);
-        };
         auto snap_dismiss = [](const Appointment&){};
-        snap(appt, snap_show, snap_dismiss);
+        snap(appt, nullptr, snap_dismiss);
     });
 
     // create the menus
