diff -Naur indicator-datetime-13.10.0+13.10.20131023.2.orig/CMakeLists.txt indicator-datetime-13.10.0+13.10.20131023.2/CMakeLists.txt
--- indicator-datetime-13.10.0+13.10.20131023.2.orig/CMakeLists.txt	2014-01-09 15:00:35.965288787 +0100
+++ indicator-datetime-13.10.0+13.10.20131023.2/CMakeLists.txt	2014-01-09 14:52:46.085214653 +0100
@@ -39,7 +39,6 @@
                    libecal-1.2>=3.5
                    libedataserver-1.2>=3.5
                    libnotify>=0.7.6
-                   url-dispatcher-1>=1
                    json-glib-1.0>=0.16.2)
 include_directories (${SERVICE_INCLUDE_DIRS})
 
diff -Naur indicator-datetime-13.10.0+13.10.20131023.2.orig/src/service.c indicator-datetime-13.10.0+13.10.20131023.2/src/service.c
--- indicator-datetime-13.10.0+13.10.20131023.2.orig/src/service.c	2014-01-09 14:56:09.085111649 +0100
+++ indicator-datetime-13.10.0+13.10.20131023.2/src/service.c	2014-01-09 14:56:30.124204483 +0100
@@ -23,7 +23,6 @@
 #include <glib/gi18n.h>
 #include <gio/gio.h>
 #include <json-glib/json-glib.h>
-#include <url-dispatcher.h>
 
 #include "dbus-shared.h"
 #include "planner-eds.h"
@@ -1353,23 +1352,6 @@
 }
 
 static void
-on_phone_settings_activated (GSimpleAction * a      G_GNUC_UNUSED,
-                             GVariant      * param  G_GNUC_UNUSED,
-                             gpointer        gself  G_GNUC_UNUSED)
-{
-  url_dispatch_send ("settings:///system/time-date", NULL, NULL);
-}
-
-static void
-on_phone_clock_activated (GSimpleAction * a      G_GNUC_UNUSED,
-                          GVariant      * param  G_GNUC_UNUSED,
-                          gpointer        gself  G_GNUC_UNUSED)
-{
-  const char * url = "appid://com.ubuntu.clock/clock/current-user-version";
-  url_dispatch_send (url, NULL, NULL);
-}
-
-static void
 on_activate_planner (GSimpleAction * a         G_GNUC_UNUSED,
                      GVariant      * param,
                      gpointer        gself)
@@ -1421,8 +1403,6 @@
 
   GActionEntry entries[] = {
     { "activate-desktop-settings", on_desktop_settings_activated },
-    { "activate-phone-settings", on_phone_settings_activated },
-    { "activate-phone-clock-app", on_phone_clock_activated },
     { "activate-planner", on_activate_planner, "x", NULL },
     { "set-location", on_set_location, "s" }
   };
