=== modified file 'Audience/Widgets/Button.vala'
--- Audience/Widgets/Button.vala	2012-05-19 15:57:37 +0000
+++ Audience/Widgets/Button.vala	2012-06-28 04:35:48 +0000
@@ -33,8 +33,11 @@
                 
                 var ti = new ThemedIcon.from_names ({icon, alt_fallback, fallback});
                 var l = Gtk.IconTheme.get_default ().lookup_by_gicon (ti, 16, 0);
-                this.set_from_pixbuf (l.load_symbolic ({1.0,1.0,1.0,1.0}, null, null, null, null));
-                
+                if (l != null) {
+                    this.set_from_pixbuf (l.load_symbolic ({1.0,1.0,1.0,1.0}, null, null, null, null));
+                } else {
+                    warning("NULL detected when trying to load icon: " + icon + " (or " + fallback + ")");
+                }                
             } catch (Error e){warning (e.message);}
         }
     }

