diff -Naur indicator-session-12.10.5+14.04.20131125.orig/data/CMakeLists.txt indicator-session-12.10.5+14.04.20131125/data/CMakeLists.txt
--- indicator-session-12.10.5+14.04.20131125.orig/data/CMakeLists.txt	2013-12-09 15:43:19.236853167 +0100
+++ indicator-session-12.10.5+14.04.20131125/data/CMakeLists.txt	2013-12-09 15:43:45.371986127 +0100
@@ -90,6 +90,4 @@
          DESTINATION "${ICON_DIR}"
          FILES_MATCHING PATTERN "*.png" PATTERN "*.svg")
 
-install(CODE "execute_process (COMMAND gtk-update-icon-cache -t -f ${ICON_DIR})"
-        CODE "message (STATUS \"Updating icon cache\")")
 
