diff -Naur indicator-session-12.10.5+14.04.20131125.orig/cmake/UseGSettings.cmake indicator-session-12.10.5+14.04.20131125/cmake/UseGSettings.cmake
--- indicator-session-12.10.5+14.04.20131125.orig/cmake/UseGSettings.cmake	2013-12-09 15:49:39.629703340 +0100
+++ indicator-session-12.10.5+14.04.20131125/cmake/UseGSettings.cmake	2013-12-09 15:51:49.914306105 +0100
@@ -17,7 +17,5 @@
     message (STATUS "${GSETTINGS_DIR} is the GSettings install dir")
     install (FILES ${SCHEMA_NAME} DESTINATION ${GSETTINGS_DIR} OPTIONAL)
 
-    install (CODE "message (STATUS \"Compiling GSettings schemas\")")
-    install (CODE "execute_process (COMMAND ${_glib_compile_schemas} ${GSETTINGS_DIR})")
 endmacro()
 
