diff -Naur indicator-session-12.10.5+14.04.20131125.orig/data/com.canonical.indicator.session.gschema.xml.in indicator-session-12.10.5+14.04.20131125/data/com.canonical.indicator.session.gschema.xml.in
--- indicator-session-12.10.5+14.04.20131125.orig/data/com.canonical.indicator.session.gschema.xml.in	2013-12-09 15:46:32.619814172 +0100
+++ indicator-session-12.10.5+14.04.20131125/data/com.canonical.indicator.session.gschema.xml.in	2013-12-09 15:47:09.904994189 +0100
@@ -1,5 +1,5 @@
 <schemalist>
-  <schema id="com.canonical.indicator.session" path="/apps/indicator-session/">
+  <schema id="com.canonical.indicator.session" path="/com/canonical/indicator/session/">
     <key name="suppress-logout-restart-shutdown" type="b">
       <default>false</default>
       <_summary>Suppress the dialog to confirm logout, restart and shutdown action</_summary>
