diff -Naur indicator-session-12.10.5+14.04.20151021.1/CMakeLists.txt indicator-session-elementary/CMakeLists.txt
--- indicator-session-12.10.5+14.04.20151021.1/CMakeLists.txt	2015-10-21 17:52:53.000000000 +0200
+++ indicator-session-elementary/CMakeLists.txt	2016-02-04 14:52:27.026675000 +0100
@@ -7,7 +7,7 @@
 set (PACKAGE ${CMAKE_PROJECT_NAME})
 set (GETTEXT_PACKAGE indicator-session)
 
-option (enable_tests "Build the package's automatic tests." ON)
+option (enable_tests "Build the package's automatic tests." OFF)
 option (enable_lcov "Generate lcov code coverage reports." ON)
 
 ##
diff -Naur indicator-session-12.10.5+14.04.20151021.1/src/backend-dbus/actions.c indicator-session-elementary/src/backend-dbus/actions.c
--- indicator-session-12.10.5+14.04.20151021.1/src/backend-dbus/actions.c	2015-10-21 17:52:53.000000000 +0200
+++ indicator-session-elementary/src/backend-dbus/actions.c	2016-02-04 14:52:27.026675000 +0100
@@ -879,7 +879,9 @@
 {
   priv_t * p = INDICATOR_SESSION_ACTIONS_DBUS(self)->priv;
 
-  if (is_owned_proxy (p->unity_session))
+  // testing against unity_session isn't enough for other session (pantheon)
+  // so we check if we have gnome screensaver too
+  if (is_owned_proxy (p->unity_session) && p->screen_saver == NULL)
     {
       if (immediate)
         {
diff -Naur indicator-session-12.10.5+14.04.20151021.1/src/service.c indicator-session-elementary/src/service.c
--- indicator-session-12.10.5+14.04.20151021.1/src/service.c	2015-10-21 17:53:01.000000000 +0200
+++ indicator-session-elementary/src/service.c	2016-02-04 15:26:57.990512330 +0100
@@ -27,9 +27,7 @@
 #define BUS_NAME "com.canonical.indicator.session"
 #define BUS_PATH "/com/canonical/indicator/session"
 
-#define ICON_DEFAULT "system-devices-panel"
-#define ICON_INFO    "system-devices-panel-information"
-#define ICON_ALERT   "system-devices-panel-alert"
+#define ICON_DEFAULT "system-devices-symbolic"
 
 G_DEFINE_TYPE (IndicatorSessionService,
                indicator_session_service,
@@ -345,7 +343,6 @@
 
   menu = g_menu_new ();
   g_menu_append (menu, _("About This Computer"), "indicator.about");
-  g_menu_append (menu, _("Ubuntu Help"), "indicator.help");
   return G_MENU_MODEL (menu);
 }
 
