diff --git a/data/meson.build b/data/meson.build
index 37eed82..6a93a8d 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -10,20 +10,3 @@ install_data(
     rename: meson.project_name() + '.gschema.xml'
 )
 
-dep_systemd = dependency('systemd', required: true)
-systemd_userunitdir = get_option('systemduserunitdir')
-if systemd_userunitdir == ''
-    systemd_userunitdir = dep_systemd.get_variable('systemduserunitdir', pkgconfig_define: ['prefix', get_option('prefix')])
-endif
-
-bindir = get_option('prefix') / get_option('bindir')
-unit_conf = configuration_data()
-unit_conf.set('bindir', bindir)
-
-configure_file(
-    input: 'dock.service.in',
-    output: meson.project_name() + '.service',
-    install: true,
-    install_dir: systemd_userunitdir,
-    configuration: unit_conf
-)
