--- /src/Dialogs/NewUserDialog.vala	2019-03-15 19:51:19.000000000 +0300
+++ /src/Dialogs/NewUserDialog.vala	2019-03-19 00:24:42.450791200 +0300
@@ -118,17 +118,23 @@
 
             if (get_permission ().allowed) {
                 try {
-                    var created_user = get_usermanager ().create_user (username, fullname, accounttype);
+                    Act.User created_user;
 
                     get_usermanager ().user_added.connect ((user) => {
-                        if (user == created_user) {
-                            created_user.set_locked (false);
+                        GLib.Idle.add (() => {
+                            if (user == created_user) {
+                                created_user.set_locked (false);
 
-                            if (password != null) {
-                                created_user.set_password (password, "");
+                                if (password != null) {
+                                    created_user.set_password (password, "");
+                                }
                             }
-                        }
+
+                            return false;
+                        });
                     });
+
+                    created_user = get_usermanager ().create_user (username, fullname, accounttype);
                 } catch (Error e) {
                     critical ("Creation of user '%s' failed".printf (username));
                 }
