--- /src/Widgets/UserListBox.vala	2019-11-16 01:25:40.000000000 +0300
+++ /src/Widgets/UserListBox.vala	2020-04-07 16:32:33.558205300 +0300
@@ -110,17 +110,38 @@
         }
 
         private static string get_display_manager () {
-            string output = "";
+            string file = "/etc/systemd/system/display-manager.service";
 
-            try {
-                //TODO: add file location for different, non-debian-based distros
-                FileUtils.get_contents ("/etc/X11/default-display-manager", out output);
-            } catch (Error e) {
-                critical (e.message);
+            if (FileUtils.test(file, FileTest.IS_REGULAR)) {
+                Regex regex = new Regex ("^ExecStart");
+                string read_content = "";
+                string output = "";
+
+                try {
+                    FileUtils.get_contents (file, out read_content);
+                } catch (Error e) {
+                    warning (e.message);
+                    return "";
+                }
+
+                var content_array = read_content.split ("\n");
+                foreach (string line in content_array) {
+                    if (regex.match (line)) {
+                        output = line;
+                        break;
+                    }
+                }
+
+                if (output.index_of ("/") != -1) {
+                    output = output.slice (output.last_index_of ("/") + 1, output.length).chomp ();
+                } else if (output.index_of ("=") != -1) {
+                    output = output.slice (output.last_index_of ("=") + 1, output.length).chomp ();
+                }
+
+                return output;
+            } else {
                 return "";
             }
-
-            return output.slice (output.last_index_of ("/") + 1, output.length).chomp ();
         }
     }
 }
