--- /src/Dialogs/NewUserDialog.vala	2019-11-16 01:25:40.000000000 +0300
+++ /src/Dialogs/NewUserDialog.vala	2020-04-07 15:29:53.826322800 +0300
@@ -113,17 +113,23 @@
 
                 if (get_permission ().allowed) {
                     try {
-                        var created_user = get_usermanager ().create_user (username, fullname, accounttype);
+                        Act.User created_user;
 
                         get_usermanager ().user_added.connect ((user) => {
-                            if (user == created_user) {
-                                created_user.set_locked (false);
+                            GLib.Idle.add (() => {
+                                if (user == created_user) {
+                                    created_user.set_locked (false);
 
-                                if (password != null) {
-                                    created_user.set_password (password, "");
+                                    if (password != null) {
+                                        created_user.set_password (password, "");
+                                    }
                                 }
-                            }
+
+                                return false;
+                            });
                         });
+
+                        created_user = get_usermanager ().create_user (username, fullname, accounttype);
                     } catch (Error e) {
                         critical ("Creation of user '%s' failed", username);
                     }
