--- /src/Views/UserSettingsView.vala	2019-11-16 01:25:40.000000000 +0300
+++ /src/Views/UserSettingsView.vala	2020-04-07 22:04:33.109820500 +0300
@@ -385,12 +385,23 @@
                 foreach (string language in languages) {
                     language_store.insert (out iter, 0);
                     language_store.set (iter, 0, language, 1, Gnome.Languages.get_language_from_code (language, null));
-                    if (user.get_language ().slice (0, 2) == language)
+                    string user_lang = user.get_language ();
+                    if (user_lang == "") {
+                        var env_lang = Environment.get_variable ("LANG");
+                        user_lang = env_lang != null ? env_lang : "en_US.UTF-8";
+                    }
+                    if (user_lang.slice (0, 2) == language) {
                         language_box.set_active_iter (iter);
+                    }
                 }
 
             } else {
-                var language = Gnome.Languages.get_language_from_code (user.get_language ().slice (0, 2), null);
+                string user_lang = user.get_language ();
+                if (user_lang == "") {
+                    var env_lang = Environment.get_variable ("LANG");
+                    user_lang = env_lang != null ? env_lang : "en_US.UTF-8";
+                }
+                var language = Gnome.Languages.get_language_from_code (user_lang.slice (0, 2), null);
                 language_button.set_label (language);
             }
         }
