--- /src/Views/UserSettingsView.vala
+++ /src/Views/UserSettingsView.vala
@@ -410,12 +410,24 @@ namespace SwitchboardPlugUserAccounts.Wi
                 foreach (string language in languages) {
                     language_store.insert (out iter, 0);
                     language_store.set (iter, 0, language, 1, Gnome.Languages.get_language_from_code (language, null));
-                    if (user.get_language ().slice (0, 2) == language)
+                    string user_lang = user.get_language ();
+                    if (user_lang == null) {
+                        var env_lang = GLib.Environment.get_variable ("LANG");
+                        user_lang = env_lang ?? "en_US.UTF-8";
+                    }
+
+                    if (user_lang.slice (0, 2) == language) {
                         language_box.set_active_iter (iter);
+                    }
                 }
 
             } else {
-                var language = Gnome.Languages.get_language_from_code (user.get_language ().slice (0, 2), null);
+                string user_lang = user.get_language ();
+                if (user_lang == null || user_lang == "") {
+                    var env_lang = Environment.get_variable ("LANG");
+                    user_lang = env_lang != null ? env_lang : "en_US.UTF-8";
+                }
+                var language = Gnome.Languages.get_language_from_code (user_lang.slice (0, 2), null);
                 language_button.set_label (language);
             }
         }
