--- /src/Views/UserSettingsView.vala
+++ /src/Views/UserSettingsView.vala
@@ -389,10 +389,14 @@ namespace SwitchboardPlugUserAccounts.Wi
 
         private Gdk.Pixbuf? avatar_image_load_func (int size) {
             try {
-                return new Gdk.Pixbuf.from_file_at_scale (user.get_icon_file (), size, size, true);
+                var icon_path = user.get_icon_file ();
+                if (GLib.FileUtils.test (icon_path, GLib.FileTest.EXISTS)) {
+                    return new Gdk.Pixbuf.from_file_at_scale (icon_path, size, size, true);
+                }
             } catch (Error e) {
-                return null;
+                warning (e.message);
             }
+            return null;
         }
 
         public void update_language () {
--- /src/Widgets/UserItem.vala
+++ /src/Widgets/UserItem.vala
@@ -98,7 +98,10 @@ namespace SwitchboardPlugUserAccounts.Wi
         }
 
         private void update_avatar_icon () {
-            avatar.set_loadable_icon (new FileIcon (File.new_for_path (user.get_icon_file ())));
+            var icon_path = user.get_icon_file ();
+            if (GLib.FileUtils.test (icon_path, GLib.FileTest.EXISTS)) {
+                avatar.set_loadable_icon (new FileIcon (File.new_for_path (icon_path)));
+            }
         }
     }
 }
