--- /src/Shortcuts/List.vala	2020-04-02 02:34:57.000000000 +0300
+++ /src/Shortcuts/List.vala	2020-04-08 00:54:02.716828200 +0300
@@ -109,7 +109,6 @@
             add_action (ref launchers_group, Schema.MEDIA, _("Email"), "email");
             add_action (ref launchers_group, Schema.MEDIA, _("Home Folder"), "home");
             add_action (ref launchers_group, Schema.MEDIA, _("Music"), "media");
-            add_action (ref launchers_group, Schema.MEDIA, _("Terminal"), "terminal");
             add_action (ref launchers_group, Schema.MEDIA, _("Internet Browser"), "www");
 
             media_group = {};
