--- /src/Shortcuts/List.vala	2019-03-06 12:03:16.000000000 +0300
+++ /src/Shortcuts/List.vala	2019-03-06 21:05:26.244470398 +0300
@@ -105,7 +105,6 @@
             add_action (ref launchers_group, Schema.MEDIA, _("Email"), "email");
             add_action (ref launchers_group, Schema.MEDIA, _("Home Folder"), "home");
             add_action (ref launchers_group, Schema.MEDIA, _("Music"), "media");
-            add_action (ref launchers_group, Schema.MEDIA, _("Terminal"), "terminal");
             add_action (ref launchers_group, Schema.MEDIA, _("Internet Browser"), "www");
 
             media_group = {};
