# Ubuntu: https://launchpad.net/bugs/842768
Index: gnome-settings-daemon-3.6.1/plugins/print-notifications/gsd-print-notifications-manager.c
===================================================================
--- gnome-settings-daemon-3.6.1.orig/plugins/print-notifications/gsd-print-notifications-manager.c	2012-10-05 13:56:44.000000000 +0200
+++ gnome-settings-daemon-3.6.1/plugins/print-notifications/gsd-print-notifications-manager.c	2012-10-26 10:19:49.487004175 +0200
@@ -293,7 +293,7 @@
         static const char * const reasons[] = {
                 "toner-low",
                 "toner-empty",
-                "connecting-to-device",
+                /*"connecting-to-device",*/
                 "cover-open",
                 "cups-missing-filter",
                 "door-open",
@@ -310,7 +310,7 @@
                 /* Translators: The printer has no toner left (same as in system-config-printer) */
                 N_("Toner empty"),
                 /* Translators: The printer is in the process of connecting to a shared network output device (same as in system-config-printer) */
-                N_("Not connected?"),
+                /*N_("Not connected?"),*/
                 /* Translators: One or more covers on the printer are open (same as in system-config-printer) */
                 N_("Cover open"),
                 /* Translators: A filter or backend is not installed (same as in system-config-printer) */
@@ -336,7 +336,7 @@
                 /* Translators: The printer has no toner left (same as in system-config-printer) */
                 N_("Printer '%s' has no toner left."),
                 /* Translators: The printer is in the process of connecting to a shared network output device (same as in system-config-printer) */
-                N_("Printer '%s' may not be connected."),
+                /*N_("Printer '%s' may not be connected."),*/
                 /* Translators: One or more covers on the printer are open (same as in system-config-printer) */
                 N_("The cover is open on printer '%s'."),
                 /* Translators: A filter or backend is not installed (same as in system-config-printer) */
