diff -burN gnome-settings-daemon-3.10.0/configure gnome-settings-daemon-3.10.0.new/configure
--- gnome-settings-daemon-3.10.0/configure	2013-09-24 05:30:12.000000000 -0500
+++ gnome-settings-daemon-3.10.0.new/configure	2013-10-05 22:09:40.678815465 -0500
@@ -666,6 +666,8 @@
 WACOM_OLED_CFLAGS
 WACOM_LIBS
 WACOM_CFLAGS
+WITH_COLOR_FALSE
+WITH_COLOR_TRUE
 GEOCLUE_DBUS_INTERFACE_XML
 DATETIME_LIBS
 DATETIME_CFLAGS
@@ -920,6 +922,8 @@
 enable_schemas_compile
 enable_gudev
 enable_ibus
+enable_color
+enable_wacom
 enable_packagekit
 enable_smartcard_support
 with_nssdb
@@ -1640,6 +1644,8 @@
   --disable-gudev         Disable GUdev support (not optional on Linux
                           platforms)
   --disable-ibus          Disable IBus support
+  --disable-color         turn off color plugin
+  --disable-wacom         turn off wacom plugin
   --disable-packagekit    turn off PackageKit support
   --disable-smartcard-support
                           turn off smartcard support
@@ -19761,6 +19767,14 @@
 fi
 
 
+# Check whether --enable-color was given.
+if test "${enable_color+set}" = set; then :
+  enableval=$enable_color; with_color=$enableval
+else
+  with_color=yes
+fi
+
+if test x$with_color = xyes; then
 
 pkg_failed=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for COLOR" >&5
@@ -19853,6 +19867,21 @@
 
 fi
 
+$as_echo "#define HAVE_COLOR 1" >>confdefs.h
+
+fi
+ if test "x$with_color" = "xyes"; then
+  WITH_COLOR_TRUE=
+  WITH_COLOR_FALSE='#'
+else
+  WITH_COLOR_TRUE='#'
+  WITH_COLOR_FALSE=
+fi
+
+
+
+
+
 
 LIBM=
 case $host in
@@ -20120,6 +20149,13 @@
       have_wacom=no
     else
       if test x$enable_gudev != xno; then
+      	 # Check whether --enable-wacom was given.
+if test "${enable_wacom+set}" = set; then :
+  enableval=$enable_wacom; with_wacom=$enableval
+else
+  with_wacom=no
+fi
+          if test x$with_wacom = xyes; then
 
 pkg_failed=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for WACOM" >&5
@@ -20302,10 +20338,14 @@
 $as_echo "yes" >&6; }
 
 fi
+                 have_wacom=yes
+	 fi
+
+
       else
         as_fn_error $? "GUdev is necessary to compile Wacom support" "$LINENO" 5
       fi
-      have_wacom=yes
+
     fi
     ;;
   *)
@@ -21095,6 +21135,10 @@
   as_fn_error $? "conditional \"HAVE_INTROSPECTION\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
+if test -z "${WITH_COLOR_TRUE}" && test -z "${WITH_COLOR_FALSE}"; then
+  as_fn_error $? "conditional \"WITH_COLOR\" was never defined.
+Usually this means the macro was only invoked conditionally." "$LINENO" 5
+fi
 if test -z "${HAVE_WACOM_TRUE}" && test -z "${HAVE_WACOM_FALSE}"; then
   as_fn_error $? "conditional \"HAVE_WACOM\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
diff -burN gnome-settings-daemon-3.10.0/configure.ac gnome-settings-daemon-3.10.0.new/configure.ac
--- gnome-settings-daemon-3.10.0/configure.ac	2013-09-24 05:25:37.000000000 -0500
+++ gnome-settings-daemon-3.10.0.new/configure.ac	2013-10-05 22:09:47.341815419 -0500
@@ -245,7 +245,19 @@
 dnl - color
 dnl ---------------------------------------------------------------------------
 
-PKG_CHECK_MODULES(COLOR, [colord >= 1.0.2 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3])
+AC_ARG_ENABLE([color],
+  AS_HELP_STRING([--disable-color],
+                 [turn off color plugin]),
+       [with_color=$enableval],
+       [with_color=yes]) dnl Default value
+if test x$with_color = xyes; then
+       PKG_CHECK_MODULES(COLOR, [colord >= 0.1.9 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3])
+       AC_DEFINE(HAVE_COLOR, 1, [Define if color plugin is enabled])
+fi                                                                                                                                                                                                                
+AM_CONDITIONAL(WITH_COLOR, test "x$with_color" = "xyes")                                                                                                                                                          
+                                                                                                                                                                                                                  
+AC_SUBST(COLOR_CFLAGS)                                                                                                                                                                                            
+AC_SUBST(COLOR_LIBS)                                          
 
 dnl ---------------------------------------------------------------------------
 dnl - datetime
@@ -276,12 +288,23 @@
       have_wacom=no
     else
       if test x$enable_gudev != xno; then
+      	 AC_ARG_ENABLE([wacom],
+	   AS_HELP_STRING([--disable-wacom],
+	                  [turn off wacom plugin]),
+	                  [with_wacom=$enableval],
+		          [with_wacom=no]) dnl Default value
+         if test x$with_wacom = xyes; then
         PKG_CHECK_MODULES(WACOM, [libwacom >= $LIBWACOM_REQUIRED_VERSION x11 xi xtst gudev-1.0 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION xorg-wacom librsvg-2.0 >= $LIBRSVG_REQUIRED_VERSION libnotify >= $LIBNOTIFY_REQUIRED_VERSION pango >= $PANGO_REQUIRED_VERSION])
         PKG_CHECK_MODULES(WACOM_OLED, [gudev-1.0])
+                have_wacom=yes
+	 else
+                have_wacom=no
+	 fi
+	AC_SUBST(WACOM_CFLAGS)
+	AC_SUBST(WACOM_LIBS)
       else
         AC_MSG_ERROR([GUdev is necessary to compile Wacom support])
       fi
-      have_wacom=yes
     fi
     ;;
   *)
diff -burN gnome-settings-daemon-3.10.0/plugins/Makefile.am gnome-settings-daemon-3.10.0.new/plugins/Makefile.am
--- gnome-settings-daemon-3.10.0/plugins/Makefile.am	2013-09-02 21:43:53.000000000 -0500
+++ gnome-settings-daemon-3.10.0.new/plugins/Makefile.am	2013-10-05 22:09:40.678815465 -0500
@@ -4,7 +4,6 @@
 	a11y-keyboard	\
 	a11y-settings	\
 	clipboard	\
-	color		\
 	cursor		\
 	datetime	\
 	dummy		\
@@ -22,6 +21,12 @@
 
 disabled_plugins = $(NULL)
 
+if WITH_COLOR
+enabled_plugins += color
+else
+disabled_plugins += color
+endif
+
 if HAVE_PACKAGEKIT
 enabled_plugins += updates
 else
