diff -Naur gnome-settings-daemon-3.10.2.orig/gnome-settings-daemon/main.c gnome-settings-daemon-3.10.2/gnome-settings-daemon/main.c
--- gnome-settings-daemon-3.10.2.orig/gnome-settings-daemon/main.c	2013-12-13 14:12:26.332291160 +0100
+++ gnome-settings-daemon-3.10.2/gnome-settings-daemon/main.c	2013-12-13 14:12:40.003356808 +0100
@@ -342,9 +342,6 @@
         GDBusProxy *proxy;
 
         proxy = G_DBUS_PROXY (gnome_settings_bus_get_session_proxy ());
-        /* Always call this first, as Setenv can only be called before
-           any client registers */
-        set_locale (proxy);
 #ifdef HAVE_IBUS
         set_legacy_ibus_env_vars (proxy);
 #endif
