--- /data/meson.build	2018-11-04 06:49:06.000000000 +0300
+++ /data/meson.build	2018-12-18 01:12:28.433261380 +0300
@@ -19,7 +19,7 @@
 
 i18n.merge_file(
     input: meson.project_name() + '.desktop.in',
-    output: meson.project_name() + '.desktop',
+    output: 'io.elementary.ciano.desktop',
     po_dir: join_paths(meson.source_root(), 'po'),
     type: 'desktop',
     install: true,
@@ -37,4 +37,4 @@
 install_data(
     meson.project_name() + '.gschema.xml',
     install_dir: join_paths(get_option('datadir'), 'glib-2.0', 'schemas')
-)
\ В конце файла нет новой строки
+)
