Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/m17n-db/+bug/191451

Index: gtk+3.0-3.3.20/modules/input/imviqr.c
===================================================================
--- gtk+3.0-3.3.20.orig/modules/input/imviqr.c	2012-03-18 21:44:22.000000000 +0100
+++ gtk+3.0-3.3.20/modules/input/imviqr.c	2012-03-20 18:48:18.101061455 +0100
@@ -242,7 +242,7 @@
   N_("Vietnamese (VIQR)"), /* Human readable name */
   GETTEXT_PACKAGE,	   /* Translation domain */
    GTK_LOCALEDIR,	   /* Dir for bindtextdomain (not strictly needed for "gtk+") */
-  "vi"			   /* Languages for which this module is the default */
+  ""			   /* Languages for which this module is the default */
 };
 
 static const GtkIMContextInfo *info_list[] = {
