--- gtk/tests/filechooser.c.orig	2013-10-18 18:26:11.247696989 +0200
+++ gtk/tests/filechooser.c	2013-10-18 18:27:31.927762072 +0200
@@ -1603,20 +1603,6 @@
       NULL,			/* final_current_folder */
       FOLDER_NAME_2		/* final_filename */
     },
-    {
-      "select-folder-dialog-4",
-      GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER,
-      FOLDER_NAME,		/* initial_current_folder */
-      NULL,			/* initial_filename */
-      TRUE,			/* open_dialog */
-      BUTTON,			/* what_to_tweak */
-      NULL,			/* tweak_current_folder */
-      FOLDER_NAME_2,		/* tweak_filename */
-      GTK_RESPONSE_ACCEPT,	/* dialog_response */
-      FALSE,			/* unselect_all */
-      NULL,			/* final_current_folder */
-      FOLDER_NAME_2		/* final_filename */
-    },
 
     /* Unselection tests */
     {
