diff -Naur gtk+-3.0.12.orig/gtk/gtkmain.c gtk+-3.0.12/gtk/gtkmain.c
--- gtk+-3.0.12.orig/gtk/gtkmain.c	2011-08-24 10:20:41.000000000 +0200
+++ gtk+-3.0.12/gtk/gtkmain.c	2011-08-24 10:21:18.000000000 +0200
@@ -125,6 +125,7 @@
 #include "gtkselectionprivate.h"
 #include "gtksettingsprivate.h"
 #include "gtkwidgetprivate.h"
+#include "gtkwidget.h"
 #include "gtkwindowprivate.h"
 #include "gtktooltip.h"
 #include "gtkdebug.h"
@@ -2193,7 +2194,7 @@
 
   if (!gtk_widget_has_grab (widget) && gtk_widget_is_sensitive (widget))
     {
-      _gtk_widget_set_has_grab (widget, TRUE);
+      gtk_widget_set_has_grab (widget, TRUE);
 
       group = gtk_main_get_window_group (widget);
 
@@ -2244,7 +2245,7 @@
 
   if (gtk_widget_has_grab (widget))
     {
-      _gtk_widget_set_has_grab (widget, FALSE);
+      gtk_widget_set_has_grab (widget, FALSE);
 
       group = gtk_main_get_window_group (widget);
       _gtk_window_group_remove_grab (group, widget);
diff -Naur gtk+-3.0.12.orig/gtk/gtk.symbols gtk+-3.0.12/gtk/gtk.symbols
--- gtk+-3.0.12.orig/gtk/gtk.symbols	2011-08-24 10:20:41.000000000 +0200
+++ gtk+-3.0.12/gtk/gtk.symbols	2011-08-24 10:21:18.000000000 +0200
@@ -3619,6 +3619,7 @@
 gtk_widget_set_double_buffered
 gtk_widget_set_events
 gtk_widget_set_halign
+gtk_widget_set_has_grab
 gtk_widget_set_has_tooltip
 gtk_widget_set_has_window
 gtk_widget_set_hexpand
diff -Naur gtk+-3.0.12.orig/gtk/gtkwidget.c gtk+-3.0.12/gtk/gtkwidget.c
--- gtk+-3.0.12.orig/gtk/gtkwidget.c	2011-08-24 10:20:41.000000000 +0200
+++ gtk+-3.0.12/gtk/gtkwidget.c	2011-08-24 10:21:18.000000000 +0200
@@ -6869,7 +6869,7 @@
 }
 
 void
-_gtk_widget_set_has_grab (GtkWidget *widget,
+gtk_widget_set_has_grab (GtkWidget *widget,
                           gboolean   has_grab)
 {
   widget->priv->has_grab = has_grab;
diff -Naur gtk+-3.0.12.orig/gtk/gtkwidget.h gtk+-3.0.12/gtk/gtkwidget.h
--- gtk+-3.0.12.orig/gtk/gtkwidget.h	2011-08-24 10:20:41.000000000 +0200
+++ gtk+-3.0.12/gtk/gtkwidget.h	2011-08-24 10:23:27.000000000 +0200
@@ -569,6 +569,8 @@
                                            gboolean             receives_default);
 gboolean  gtk_widget_get_receives_default (GtkWidget           *widget);
 
+void       gtk_widget_set_has_grab        (GtkWidget           *widget,
+                                           gboolean             has_grab);
 gboolean   gtk_widget_has_grab            (GtkWidget           *widget);
 
 gboolean   gtk_widget_device_is_shadowed  (GtkWidget           *widget,
@@ -942,8 +944,6 @@
                                                            gboolean      clip_to_size);
 void              _gtk_widget_set_has_default             (GtkWidget    *widget,
                                                            gboolean      has_default);
-void              _gtk_widget_set_has_grab                (GtkWidget    *widget,
-                                                           gboolean      has_grab);
 void              _gtk_widget_set_is_toplevel             (GtkWidget    *widget,
                                                            gboolean      is_toplevel);
 
