diff --git a/gdk/wayland/gdkdevice-wayland.c b/gdk/wayland/gdkdevice-wayland.c
index 9e8e6b1..237a5b8 100644
--- a/gdk/wayland/gdkdevice-wayland.c
+++ b/gdk/wayland/gdkdevice-wayland.c
@@ -32,7 +32,7 @@
 #include "gdkdevicemanagerprivate.h"
 #include "gdkprivate-wayland.h"
 
-#include <X11/extensions/XKBcommon.h>
+#include <xkbcommon/xkbcommon.h>
 #include <X11/keysym.h>
 
 #include <sys/time.h>
diff --git a/gdk/wayland/gdkdisplaymanager-wayland.c b/gdk/wayland/gdkdisplaymanager-wayland.c
index eca6f92..d59563a 100644
--- a/gdk/wayland/gdkdisplaymanager-wayland.c
+++ b/gdk/wayland/gdkdisplaymanager-wayland.c
@@ -27,7 +27,7 @@
 
 #include "gdkinternals.h"
 
-#include <X11/extensions/XKBcommon.h>
+#include <xkbcommon/xkbcommon.h>
 
 typedef struct _GdkWaylandDisplayManager GdkWaylandDisplayManager;
 typedef struct _GdkWaylandDisplayManagerClass GdkWaylandDisplayManagerClass;
diff --git a/gdk/wayland/gdkkeys-wayland.c b/gdk/wayland/gdkkeys-wayland.c
index f3e9d3c..bc26a43 100644
--- a/gdk/wayland/gdkkeys-wayland.c
+++ b/gdk/wayland/gdkkeys-wayland.c
@@ -39,7 +39,7 @@
 #include "gdkdisplay-wayland.h"
 #include "gdkkeysprivate.h"
 
-#include <X11/extensions/XKBcommon.h>
+#include <xkbcommon/xkbcommon.h>
 
 typedef struct _GdkWaylandKeymap          GdkWaylandKeymap;
 typedef struct _GdkWaylandKeymapClass     GdkWaylandKeymapClass;
