Bug-Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=684533

--- a/modules/printbackends/cups/gtkprintbackendcups.c
+++ b/modules/printbackends/cups/gtkprintbackendcups.c
@@ -1896,6 +1896,19 @@
 		   &port,
 		   resource, sizeof (resource));
 
+  if (strncmp (method, "dnssd", 5) == 0)
+    {
+      _httpResolveURI(cups_printer->printer_uri,
+		      uri, sizeof(uri), 0, NULL, NULL);
+      httpSeparateURI (HTTP_URI_CODING_ALL,
+		       uri,
+		       method, sizeof (method),
+		       username, sizeof (username),
+		       hostname, sizeof (hostname),
+		       &port,
+		       resource, sizeof (resource));
+    }
+
   if (strncmp (resource, "/printers/", 10) == 0)
     {
       cups_printer->ppd_name = g_strdup (resource + 10);
