diff -Naur src/Makefile.am src/Makefile.am
--- src/Makefile.am	2011-08-21 00:19:38.000000000 +0200
+++ src/Makefile.am	2011-08-21 00:20:36.000000000 +0200
@@ -48,7 +48,7 @@
 	-DLIBDIR=\"$(libdir)"\"		\
 	-DG_DISABLE_DEPRECATED		\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGDK_DISABLE_DEPRECATED
+	-DGDK_NODISABLE_DEPRECATED
 
 AM_CPPFLAGS =				\
 	$(GCC_FLAGS)			\
diff -Naur src/idoscalemenuitem.c src/idoscalemenuitem.c
--- src/idoscalemenuitem.c	2011-08-21 00:19:38.000000000 +0200
+++ src/idoscalemenuitem.c	2011-08-21 00:21:51.000000000 +0200
@@ -480,12 +480,12 @@
   translate_event_coordinates (menuitem, event->x_root, &x);
   event->x_root = x;
 
-  ubuntu_gtk_widget_set_has_grab (scale, TRUE);
+  GTK_OBJECT_FLAGS (scale) |= GTK_HAS_GRAB;
 
   gtk_widget_event (scale,
                     ((GdkEvent *)(void*)(event)));
 
-  ubuntu_gtk_widget_set_has_grab (scale, FALSE);
+  GTK_OBJECT_FLAGS (scale) &= ~(GTK_HAS_GRAB);
 
   if (!priv->grabbed)
     {
