diff -Naur ido-12.10.3daily13.03.01.orig/src/idoscalemenuitem.c ido-12.10.3daily13.03.01/src/idoscalemenuitem.c
--- ido-12.10.3daily13.03.01.orig/src/idoscalemenuitem.c	2013-11-28 16:59:00.534417271 +0100
+++ ido-12.10.3daily13.03.01/src/idoscalemenuitem.c	2013-11-28 17:10:25.762664756 +0100
@@ -182,8 +182,8 @@
       priv->right_padding = primary_padding;
     }
 
-  priv->child_allocation.x = gtk_container_get_border_width (GTK_CONTAINER (widget)) + gtk_widget_get_style (widget)->xthickness;
-  priv->child_allocation.y = gtk_container_get_border_width (GTK_CONTAINER (widget)) + gtk_widget_get_style (widget)->ythickness;
+  priv->child_allocation.x = gtk_container_get_border_width (GTK_CONTAINER (widget));
+  priv->child_allocation.y = gtk_container_get_border_width (GTK_CONTAINER (widget));
 
   priv->child_allocation.x += horizontal_padding;
   priv->child_allocation.x += priv->toggle_size;
