diff -Naur indicator-bluetooth-0.0.6+14.04.20140124.orig/data/Makefile.am indicator-bluetooth-0.0.6+14.04.20140124/data/Makefile.am
--- indicator-bluetooth-0.0.6+14.04.20140124.orig/data/Makefile.am	2014-02-06 12:00:44.362431397 +0100
+++ indicator-bluetooth-0.0.6+14.04.20140124/data/Makefile.am	2014-02-06 12:02:02.982825517 +0100
@@ -6,27 +6,6 @@
 indicatorsdir = $(datadir)/unity/indicators
 dist_indicators_DATA = com.canonical.indicator.bluetooth
 
-# the upstart job file
-upstart_jobsdir = $(datadir)/upstart/sessions
-upstart_jobs_DATA = indicator-bluetooth.conf
-upstart_jobs_in = $(upstart_jobs_DATA:.conf=.conf.in)
-$(upstart_jobs_DATA): $(upstart_jobs_in)
-	$(AM_V_GEN) $(SED) -e "s|\@pkglibexecdir\@|$(pkglibexecdir)|" $< > $@
-BUILT_SOURCES += $(upstart_jobs_DATA)
-EXTRA_DIST += $(upstart_jobs_in)
-CLEANFILES += $(upstart_jobs_DATA)
-
-# the upstart version of the xdg autostart job file
-# see https://code.launchpad.net/~charlesk/indicator-bluetooth/upstart-job/+merge/198100
-upstart_xdg_autostartdir = $(datadir)/upstart/xdg/autostart
-upstart_xdg_autostart_DATA = indicator-bluetooth.desktop
-upstart_xdg_autostart_in = indicator-bluetooth.upstart.desktop.in
-$(upstart_xdg_autostart_DATA): $(upstart_xdg_autostart_in)
-	$(AM_V_GEN) $(SED) -e "s|\@pkglibexecdir\@|$(pkglibexecdir)|" $< > $@
-BUILT_SOURCES += $(upstart_xdg_autostart_DATA)
-EXTRA_DIST += $(upstart_xdg_autostart_in)
-CLEANFILES += $(upstart_xdg_autostart_DATA)
-
 # the xdg autostart job file
 xdg_autostartdir = /etc/xdg/autostart
 xdg_autostart_DATA = indicator-bluetooth.desktop
@@ -45,8 +24,7 @@
 EXTRA_DIST += \
   com.canonical.indicator.bluetooth.gschema.xml.in \
   indicator-bluetooth.conf.in \
-  indicator-bluetooth.desktop.in \
-  indicator-bluetooth.upstart.desktop.in
+  indicator-bluetooth.desktop.in
 
 CLEANFILES += \
   $(dbus_services_DATA) \
