diff -Naur indicator-bluetooth-0.0.6+14.04.20131029.1.orig/configure.ac indicator-bluetooth-0.0.6+14.04.20131029.1/configure.ac
--- indicator-bluetooth-0.0.6+14.04.20131029.1.orig/configure.ac	2013-12-03 14:49:12.800242417 +0100
+++ indicator-bluetooth-0.0.6+14.04.20131029.1/configure.ac	2013-12-03 14:50:08.768582668 +0100
@@ -16,11 +16,9 @@
 
 GLIB_REQUIRED_VERSION=2.36
 GIO_REQUIRED_VERSION=2.36
-URL_DISPATCHER_1_REQUIRED_VERSION=1
 
 PKG_CHECK_MODULES(SERVICE_DEPS, [glib-2.0 >= $GLIB_REQUIRED_VERSION
-                                 gio-unix-2.0 >= $GIO_REQUIRED_VERSION
-                                 url-dispatcher-1 >= $URL_DISPATCHER_1_REQUIRED_VERSION])
+                                 gio-unix-2.0 >= $GIO_REQUIRED_VERSION])
 
 
 dnl ##############################
diff -Naur indicator-bluetooth-0.0.6+14.04.20131029.1.orig/src/Makefile.am indicator-bluetooth-0.0.6+14.04.20131029.1/src/Makefile.am
--- indicator-bluetooth-0.0.6+14.04.20131029.1.orig/src/Makefile.am	2013-12-03 14:49:12.800242417 +0100
+++ indicator-bluetooth-0.0.6+14.04.20131029.1/src/Makefile.am	2013-12-03 14:52:14.562345487 +0100
@@ -7,7 +7,6 @@
 	desktop.vala \
 	device.vala \
 	main.vala \
-	phone.vala \
 	profile.vala \
 	killswitch.vala \
 	service.vala
@@ -18,7 +17,6 @@
 	--vapidir=./ \
 	--pkg config \
 	--pkg rfkill \
-	--pkg url-dispatcher \
 	--pkg posix \
 	--pkg glib-2.0 \
 	--pkg gio-2.0
diff -Naur indicator-bluetooth-0.0.6+14.04.20131029.1.orig/src/phone.vala indicator-bluetooth-0.0.6+14.04.20131029.1/src/phone.vala
--- indicator-bluetooth-0.0.6+14.04.20131029.1.orig/src/phone.vala	2013-12-03 14:49:12.800242417 +0100
+++ indicator-bluetooth-0.0.6+14.04.20131029.1/src/phone.vala	1970-01-01 01:00:00.000000000 +0100
@@ -1,77 +0,0 @@
-/*
- * Copyright 2013 Canonical Ltd.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; version 3.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- *
- * Authors:
- *   Charles Kerr <charles.kerr@canonical.com>
- */
-
-class Phone: Profile
-{
-  SimpleActionGroup action_group;
-
-  public Phone (Bluetooth bluetooth, SimpleActionGroup action_group)
-  {
-    const string profile_name = "phone";
-    base (bluetooth, profile_name);
-
-    this.bluetooth = bluetooth;
-    this.action_group = action_group;
-
-    // build the static actions
-    Action[] actions = {};
-    actions += root_action;
-    actions += create_supported_action (bluetooth);
-    actions += create_enabled_action (bluetooth);
-    actions += create_settings_action ();
-    foreach (var a in actions)
-      action_group.add_action (a);
-
-    var section = new Menu ();
-    section.append_item (create_enabled_menuitem ());
-    section.append (_("Bluetooth settings…"),
-                    "indicator.phone-show-settings::bluetooth");
-    menu.append_section (null, section);
-
-    // know when to show the indicator & when to hide it
-    bluetooth.notify.connect (() => update_visibility());
-    update_visibility ();
-
-    bluetooth.notify.connect (() => update_root_action_state());
-  }
-
-  void update_visibility ()
-  {
-    visible = bluetooth.enabled;
-  }
-
-  ///
-  ///  Actions
-  ///
-
-  void show_settings (string panel)
-  {
-    UrlDispatch.send ("settings:///system/bluetooth");
-  }
-
-  Action create_settings_action ()
-  {
-    var action = new SimpleAction ("phone-show-settings", VariantType.STRING);
-
-    action.activate.connect ((action, panel)
-        => show_settings (panel.get_string()));
-
-    return action;
-  }
-}
diff -Naur indicator-bluetooth-0.0.6+14.04.20131029.1.orig/src/service.vala indicator-bluetooth-0.0.6+14.04.20131029.1/src/service.vala
--- indicator-bluetooth-0.0.6+14.04.20131029.1.orig/src/service.vala	2013-12-03 14:49:12.800242417 +0100
+++ indicator-bluetooth-0.0.6+14.04.20131029.1/src/service.vala	2013-12-03 14:52:47.186542916 +0100
@@ -52,7 +52,6 @@
     actions = new SimpleActionGroup ();
 
     profiles = new HashTable<string,Profile> (str_hash, str_equal);
-    profiles.insert ("phone", new Phone (bluetooth, actions));
     profiles.insert ("desktop", new Desktop (bluetooth, actions));
   }
 
diff -Naur indicator-bluetooth-0.0.6+14.04.20131029.1.orig/vapi/url-dispatcher.vapi indicator-bluetooth-0.0.6+14.04.20131029.1/vapi/url-dispatcher.vapi
--- indicator-bluetooth-0.0.6+14.04.20131029.1.orig/vapi/url-dispatcher.vapi	2013-12-03 14:49:12.800242417 +0100
+++ indicator-bluetooth-0.0.6+14.04.20131029.1/vapi/url-dispatcher.vapi	1970-01-01 01:00:00.000000000 +0100
@@ -1,9 +0,0 @@
-[CCode (cprefix = "", lower_case_cprefix = "", cheader_filename = "liburl-dispatcher-1/url-dispatcher.h")]
-
-namespace UrlDispatch
-{
-  public delegate void DispatchCallback ();
-
-  [CCode (cname = "url_dispatch_send")]
-  public static void send (string url, DispatchCallback? func = null);
-}
